--- a/plugins/cursor/gsd-cursor-manager.c
+++ b/plugins/cursor/gsd-cursor-manager.c
@@ -443,7 +443,8 @@
         }
 
         /* Start by hiding the cursor */
-        set_cursor_visibility (manager, FALSE);
+        g_debug ("This is a build for Unity. The cursor will NOT be hidden initially");
+        set_cursor_visibility (manager, TRUE);
 
         manager->priv->cancellable = g_cancellable_new ();
         register_manager_dbus (manager);
