diff -Nru gnome-settings-daemon-3.5.92.orig/plugins/power/gsd-power-manager.c gnome-settings-daemon-3.5.92/plugins/power/gsd-power-manager.c
--- gnome-settings-daemon-3.5.92.orig/plugins/power/gsd-power-manager.c	2012-09-20 23:35:59.563132725 -0400
+++ gnome-settings-daemon-3.5.92/plugins/power/gsd-power-manager.c	2012-09-20 23:37:47.547067749 -0400
@@ -690,7 +690,7 @@
 
         /* show a different icon if we are disconnected */
         icon = engine_get_icon (manager);
-        gtk_status_icon_set_visible (manager->priv->status_icon, icon != NULL);
+        gtk_status_icon_set_visible (manager->priv->status_icon, FALSE);
 
         if (icon == NULL) {
                 /* none before, now none */
@@ -706,6 +706,7 @@
         if (manager->priv->previous_icon == NULL) {
 
                 /* set fallback icon */
+                gtk_status_icon_set_visible (manager->priv->status_icon, FALSE);
                 gtk_status_icon_set_from_gicon (manager->priv->status_icon, icon);
                 manager->priv->previous_icon = icon;
                 return TRUE;
@@ -3690,6 +3691,7 @@
         /* TRANSLATORS: this is the title of the power manager status icon
          * that is only shown in fallback mode */
         gtk_status_icon_set_title (manager->priv->status_icon, _("Power Manager"));
+        gtk_status_icon_set_visible (manager->priv->status_icon, FALSE);
 
         /* connect to UPower for async power operations */
         g_dbus_proxy_new_for_bus (G_BUS_TYPE_SYSTEM,
