diff -Nru gnome-settings-daemon-3.5.92.orig/plugins/media-keys/gsd-media-keys-manager.c gnome-settings-daemon-3.5.92/plugins/media-keys/gsd-media-keys-manager.c
--- gnome-settings-daemon-3.5.92.orig/plugins/media-keys/gsd-media-keys-manager.c	2012-09-20 23:26:57.276347128 -0400
+++ gnome-settings-daemon-3.5.92/plugins/media-keys/gsd-media-keys-manager.c	2012-09-21 00:10:39.665347161 -0400
@@ -924,6 +924,12 @@
 }
 
 static void
+do_logout_action (GsdMediaKeysManager *manager)
+{
+        execute (manager, "gnome-session-quit --logout", FALSE);
+}
+
+static void
 do_eject_action_cb (GDrive              *drive,
                     GAsyncResult        *res,
                     GsdMediaKeysManager *manager)
@@ -2045,7 +2051,7 @@
                 do_sound_action (manager, deviceid, VOLUME_UP_KEY, TRUE);
                 break;
         case LOGOUT_KEY:
-                gnome_session_shutdown (manager);
+                do_logout_action (manager);
                 break;
         case EJECT_KEY:
                 do_eject_action (manager);
