diff -Nru gnome-settings-daemon-3.5.92.orig/plugins/power/gsd-power-manager.c gnome-settings-daemon-3.5.92/plugins/power/gsd-power-manager.c
--- gnome-settings-daemon-3.5.92.orig/plugins/power/gsd-power-manager.c	2012-09-20 23:39:35.536031220 -0400
+++ gnome-settings-daemon-3.5.92/plugins/power/gsd-power-manager.c	2012-09-21 00:11:49.996316135 -0400
@@ -1636,7 +1636,7 @@
                              get_first_themed_icon_name (icon),
                              &manager->priv->notification_low);
         notify_notification_set_timeout (manager->priv->notification_low,
-                                         GSD_POWER_MANAGER_NOTIFY_TIMEOUT_NEVER);
+                                         GSD_POWER_MANAGER_NOTIFY_TIMEOUT_LONG);
         notify_notification_set_urgency (manager->priv->notification_low,
                                          NOTIFY_URGENCY_CRITICAL);
         notify_notification_set_app_name (manager->priv->notification_low, _("Power"));
@@ -1783,7 +1783,7 @@
                              get_first_themed_icon_name (icon),
                              &manager->priv->notification_low);
         notify_notification_set_timeout (manager->priv->notification_low,
-                                         GSD_POWER_MANAGER_NOTIFY_TIMEOUT_NEVER);
+                                         GSD_POWER_MANAGER_NOTIFY_TIMEOUT_LONG);
         notify_notification_set_urgency (manager->priv->notification_low,
                                          NOTIFY_URGENCY_CRITICAL);
         notify_notification_set_app_name (manager->priv->notification_low, _("Power"));
