diff -Nru gnome-settings-daemon-3.5.92.orig/plugins/power/gsd-power-manager.c gnome-settings-daemon-3.5.92/plugins/power/gsd-power-manager.c
--- gnome-settings-daemon-3.5.92.orig/plugins/power/gsd-power-manager.c	2012-09-21 00:12:17.843303897 -0400
+++ gnome-settings-daemon-3.5.92/plugins/power/gsd-power-manager.c	2012-09-21 00:12:33.825297775 -0400
@@ -1167,7 +1167,7 @@
 
         /* add to database */
         array = up_client_get_devices (manager->priv->up_client);
-        for (i=0;i<array->len;i++) {
+        for (i = 0; array != NULL && i < array->len; i++) {
                 device = g_ptr_array_index (array, i);
                 engine_device_add (manager, device);
                 engine_check_recall (manager, device);
