--- a/data/org.gnome.settings-daemon.plugins.media-keys.gschema.xml.in.in
+++ b/data/org.gnome.settings-daemon.plugins.media-keys.gschema.xml.in.in
@@ -135,6 +135,11 @@
       <_summary>Record a short video of the screen</_summary>
       <description>Binding to record a short video of the screen</description>
     </key>
+    <key name="terminal" type="s">
+      <default>'&lt;Primary&gt;&lt;Alt&gt;t'</default>
+      <_summary>Launch terminal</_summary>
+      <_description>Binding to launch the terminal.</_description>
+    </key>
     <key name="www" type="s">
       <default>'XF86WWW'</default>
       <_summary>Launch web browser</_summary>
--- a/plugins/media-keys/gsd-media-keys-manager.c
+++ b/plugins/media-keys/gsd-media-keys-manager.c
@@ -979,6 +979,22 @@
 }
 
 static void
+do_terminal_action (GsdMediaKeysManager *manager)
+{
+        GSettings *settings;
+        char *term;
+
+        settings = g_settings_new ("org.gnome.desktop.default-applications.terminal");
+        term = g_settings_get_string (settings, "exec");
+
+        if (term)
+        execute (manager, term, FALSE);
+
+        g_free (term);
+        g_object_unref (settings);
+}
+
+static void
 gnome_session_shutdown (GsdMediaKeysManager *manager)
 {
 	GError *error = NULL;
@@ -2319,6 +2335,9 @@
         case SCREENCAST_KEY:
                 do_screencast_action (manager);
                 break;
+        case TERMINAL_KEY:
+                do_terminal_action (manager);
+                break;
         case WWW_KEY:
                 do_url_action (manager, "http", timestamp);
                 break;
--- a/plugins/media-keys/shortcuts-list.h
+++ b/plugins/media-keys/shortcuts-list.h
@@ -52,6 +52,7 @@
         WINDOW_SCREENSHOT_CLIP_KEY,
         AREA_SCREENSHOT_CLIP_KEY,
         SCREENCAST_KEY,
+        TERMINAL_KEY,
         WWW_KEY,
         PLAY_KEY,
         PAUSE_KEY,
@@ -126,6 +127,7 @@
         { WINDOW_SCREENSHOT_CLIP_KEY, "window-screenshot-clip", NULL, NULL, SHELL_KEYBINDING_MODE_NORMAL },
         { AREA_SCREENSHOT_CLIP_KEY, "area-screenshot-clip", NULL, NULL, SHELL_KEYBINDING_MODE_ALL },
         { SCREENCAST_KEY, "screencast", NULL, NULL, SHELL_KEYBINDING_MODE_ALL },
+        { TERMINAL_KEY, "terminal", NULL, NULL, GSD_KEYBINDING_MODE_LAUNCHER },
         { WWW_KEY, "www", NULL, NULL, GSD_KEYBINDING_MODE_LAUNCHER },
         { PLAY_KEY, "play", NULL, NULL, SHELL_KEYBINDING_MODE_ALL },
         { PAUSE_KEY, "pause", NULL, NULL, SHELL_KEYBINDING_MODE_ALL },
