diff -Nru nux-3.0.0.orig/configure.ac nux-3.0.0/configure.ac
--- nux-3.0.0.orig/configure.ac	2012-07-27 11:01:22.943658763 -0700
+++ nux-3.0.0/configure.ac	2012-07-27 11:01:59.534701440 -0700
@@ -232,18 +232,6 @@
 
 dnl ===========================================================================
 
-# sets up doxygen
-DX_DOXYGEN_FEATURE(ON)
-DX_HTML_FEATURE(ON)
-DX_CHM_FEATURE(OFF)
-DX_CHI_FEATURE(OFF)
-DX_MAN_FEATURE(OFF)
-DX_RTF_FEATURE(OFF)
-DX_XML_FEATURE(OFF)
-DX_PDF_FEATURE(OFF)
-DX_PS_FEATURE(OFF)
-DX_INIT_DOXYGEN(nux, doxygen.cfg, doc)
-
 AC_ARG_ENABLE([documentation],
               [AC_HELP_STRING([--enable-documentation=@<:@no/yes@:>@],
               [Enable building of documentation @<:@default=no@:>@])],
diff -Nru nux-3.0.0.orig/Makefile.am nux-3.0.0/Makefile.am
--- nux-3.0.0.orig/Makefile.am	2012-07-27 11:01:22.943658763 -0700
+++ nux-3.0.0/Makefile.am	2012-07-27 11:01:27.884809184 -0700
@@ -2,8 +2,6 @@
 
 CXXFLAGS += -fno-permissive
 
-include doxygen-include.am
-
 if BUILD_DOCUMENTATION
 all: doxygen-doc
 endif
