diff -Nru nux-3.0.0.orig/configure.ac nux-3.0.0/configure.ac
--- nux-3.0.0.orig/configure.ac	2012-08-02 01:03:47.303104611 -0400
+++ nux-3.0.0/configure.ac	2012-08-02 01:03:52.567106765 -0400
@@ -115,7 +115,7 @@
 PKG_PROG_PKG_CONFIG
 
 dnl ===========================================================================
-GL_PKGS="gl glu glew glewmx"
+GL_PKGS="gl glu"
 MAINTAINER_CFLAGS=""
 
 # Enable opengl es 20 code path
diff -Nru nux-3.0.0.orig/NuxGraphics/Makefile.am nux-3.0.0/NuxGraphics/Makefile.am
--- nux-3.0.0.orig/NuxGraphics/Makefile.am	2012-08-02 01:03:47.308104612 -0400
+++ nux-3.0.0/NuxGraphics/Makefile.am	2012-08-02 01:04:02.128110688 -0400
@@ -9,6 +9,7 @@
 libnux_graphics_@NUX_API_VERSION@_la_CPPFLAGS= \
   -I$(srcdir) \
   -I$(top_srcdir) \
+  -I${top_srcdir}/glew-1.7.0/include/GL \
   -DPREFIX=\""$(prefix)"\" \
 	-DLIBDIR=\""$(libdir)"\" \
   -DDATADIR=\""$(datadir)"\" \
@@ -20,6 +21,7 @@
 
 libnux_graphics_@NUX_API_VERSION@_la_LIBADD = \
   $(top_builddir)/NuxCore/libnux-core-@NUX_API_VERSION@.la \
+  ${top_srcdir}/glew-1.7.0/lib/libGLEWmx.a \
   $(NUX_GRAPHICS_LIBS)
 
 libnux_graphics_@NUX_API_VERSION@_la_LDFLAGS = \
