Index: ubuntu/tools/unity_support_test.c
===================================================================
--- ubuntu.orig/tools/unity_support_test.c	2012-01-13 09:01:58.714180000 +0100
+++ ubuntu/tools/unity_support_test.c	2012-02-27 11:16:53.435617058 +0100
@@ -587,6 +587,7 @@
   if (results->renderer != NULL &&
       (strncmp (results->renderer, "Software Rasterizer", 19) == 0 ||
        strncmp (results->renderer, "Mesa X11", 8) == 0 ||
+       strstr (results->renderer, "llvmpipe") != NULL ||
        strstr (results->renderer, "on softpipe") != NULL)) {
     results->flags |= FLAG_SOFTWARE_RENDERING;
   }
