diff -rNu unity-2d-4.8.0.orig/launcher/app/gesturehandler.cpp unity-2d-4.8.0/launcher/app/gesturehandler.cpp
--- unity-2d-4.8.0.orig/launcher/app/gesturehandler.cpp	2011-09-17 11:15:13.525609301 +0200
+++ unity-2d-4.8.0/launcher/app/gesturehandler.cpp	2011-09-17 11:17:19.659812097 +0200
@@ -25,6 +25,8 @@
 
 #include <debug_p.h>
 
+bool geis_initialized = false;
+
 GestureHandler::GestureHandler(Unity2dPanel* launcher, QObject *parent) : QObject(parent), m_launcher(launcher)
 {
     if (geisInitialize() != GEIS_STATUS_SUCCESS) {
@@ -40,11 +42,13 @@
     if (geisSubscribeGestures() != GEIS_STATUS_SUCCESS) {
         UQ_WARNING << "Subscribing to gestures failed: multitouch support disabled";
     }
+	geis_initialized = true;
 }
 
 GestureHandler::~GestureHandler()
 {
-    geis_finish(m_geisInstance);
+    if (geis_initialized)
+     geis_finish(m_geisInstance);
 }
 
 GeisStatus GestureHandler::geisInitialize()
