diff -Nru unity-2d-5.7.0.orig/spread/app/spreadmanager.cpp unity-2d-5.7.0/spread/app/spreadmanager.cpp
--- unity-2d-5.7.0.orig/spread/app/spreadmanager.cpp	2012-03-22 21:56:38.180136061 -0400
+++ unity-2d-5.7.0/spread/app/spreadmanager.cpp	2012-03-22 21:57:06.996803306 -0400
@@ -27,6 +27,7 @@
 #include <QDeclarativeEngine>
 #include <QDeclarativeItem>
 #include <QDesktopWidget>
+#include <QMouseEvent>
 
 SpreadManager::SpreadManager(QObject* parent)
  : QObject(parent)
