diff -Nru unity-2d-5.7.0.orig/libunity-2d-private/src/pointerbarrier.cpp unity-2d-5.7.0/libunity-2d-private/src/pointerbarrier.cpp
--- unity-2d-5.7.0.orig/libunity-2d-private/src/pointerbarrier.cpp	2012-03-23 14:51:14.614420799 -0400
+++ unity-2d-5.7.0/libunity-2d-private/src/pointerbarrier.cpp	2012-03-23 14:51:42.237754753 -0400
@@ -171,7 +171,7 @@
                     m_threshold,
                     0,
                     NULL);
-    Q_ASSERT(m_barrier != 0);
+    //Q_ASSERT(m_barrier != 0);
 }
 
 void PointerBarrierWrapper::destroyBarrier()
