diff -Nru network-manager-applet-0.9.4.1.bak/m4/compiler_warnings.m4 network-manager-applet-0.9.4.1/m4/compiler_warnings.m4
--- network-manager-applet-0.9.4.1.bak/m4/compiler_warnings.m4	2012-04-03 20:24:53.143443842 -0400
+++ network-manager-applet-0.9.4.1/m4/compiler_warnings.m4	2012-04-03 20:25:29.366777989 -0400
@@ -26,7 +26,7 @@
 	done
 	unset option
 	if test "x$set_more_warnings" = xerror; then
-		CFLAGS="$CFLAGS -Werror"
+		CFLAGS="$CFLAGS"
 	fi
 else
 	AC_MSG_RESULT(no)
