=== modified file 'CMakeLists.txt'
--- CMakeLists.txt	2013-03-07 13:51:41 +0000
+++ CMakeLists.txt	2013-03-12 06:43:29 +0000
@@ -15,19 +15,8 @@
 # Authored by: Thomas Voss <thomas.voss@canonical.com>,
 #              Alan Griffiths <alan@octopull.co.uk>
 
-set(
-  MIR_GCC_VERSION
-  4.7
-  CACHE
-  STRING
-  "Version of gcc/g++"
-)
 cmake_policy(SET CMP0015 NEW)
 
-set(CMAKE_C_COMPILER gcc-${MIR_GCC_VERSION})
-set(CMAKE_CXX_COMPILER g++-${MIR_GCC_VERSION})
-set(CMAKE_GCOV gcov-${MIR_GCC_VERSION})
-
 project(Mir)
 
 cmake_minimum_required(VERSION 2.8)

