From cd0973a272d9db78b535f30cf87f46fd7321ee29 Mon Sep 17 00:00:00 2001
From: Xiao-Long Chen <chenxiaolong@cxl.epac.to>
Date: Sat, 9 Aug 2014 14:56:18 -0400
Subject: [PATCH 1/2] Not a phone

---
 CMakeLists.txt | 3 +--
 src/service.c  | 2 --
 2 files changed, 1 insertion(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 569100d..30be771 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -37,8 +37,7 @@ pkg_check_modules(SERVICE_DEPS REQUIRED
                   gio-2.0>=2.36
                   gio-unix-2.0>=2.36
                   gudev-1.0>=204
-                  libnotify>=0.7.6
-                  url-dispatcher-1>=1)
+                  libnotify>=0.7.6)
 
 include_directories (SYSTEM ${SERVICE_DEPS_INCLUDE_DIRS})
 
diff --git a/src/service.c b/src/service.c
index 0cd448b..2bcbc22 100644
--- a/src/service.c
+++ b/src/service.c
@@ -20,7 +20,6 @@
 
 #include <glib/gi18n.h>
 #include <gio/gio.h>
-#include <url-dispatcher.h>
 
 #include "dbus-shared.h"
 #include "device.h"
@@ -745,7 +744,6 @@ on_phone_settings_activated (GSimpleAction * a      G_GNUC_UNUSED,
                              GVariant      * param  G_GNUC_UNUSED,
                              gpointer        gself  G_GNUC_UNUSED)
 {
-  url_dispatch_send("settings:///system/battery", NULL, NULL);
 }
 
 /***
-- 
2.0.4

