--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -42,7 +42,7 @@
 TESTS += test-device
 check_PROGRAMS += test-device
 test_device_SOURCES = test-device.cc
-test_device_LDADD = $(TEST_LIBS)
+test_device_LDADD = $(TEST_LIBS) -lpthread
 test_device_CPPFLAGS = $(TEST_CPPFLAGS)
 
 #TESTS += test-dbus-listener
@@ -54,7 +54,7 @@
 TESTS += test-indicator
 check_PROGRAMS += test-indicator
 test_indicator_SOURCES = test-indicator.cc
-test_indicator_LDADD = $(TEST_LIBS)
+test_indicator_LDADD = $(TEST_LIBS) -lpthread
 test_indicator_CPPFLAGS = $(TEST_CPPFLAGS) -DSCHEMA_DIR="\"$(top_builddir)/tests/\""
 
 ###
