--- a/configure.ac
+++ b/configure.ac
@@ -40,8 +40,7 @@
 PKG_CHECK_MODULES([SERVICE_DEPS],[glib-2.0 >= $GLIB_REQUIRED_VERSION
                                   gio-2.0 >= $GIO_REQUIRED_VERSION
                                   gio-unix-2.0 >= $GIO_UNIX_REQUIRED_VERSION
-                                  gudev-1.0 >= $GUDEV_REQUIRED_VERSION
-                                  url-dispatcher-1])
+                                  gudev-1.0 >= $GUDEV_REQUIRED_VERSION])
 
 ###########################
 # GSETTINGS
--- a/src/service.c
+++ b/src/service.c
@@ -22,7 +22,6 @@
 
 #include <glib/gi18n.h>
 #include <gio/gio.h>
-#include <url-dispatcher.h>
 
 #include "device.h"
 #include "device-provider.h"
@@ -695,7 +694,6 @@
                              GVariant      * param  G_GNUC_UNUSED,
                              gpointer        gself  G_GNUC_UNUSED)
 {
-  url_dispatch_send("settings:///system/battery", NULL, NULL);
 }
 
 /***
