=== modified file 'libindicator/indicator-image-helper.c'
--- libindicator/indicator-image-helper.c	2013-05-29 15:25:00 +0000
+++ libindicator/indicator-image-helper.c	2013-05-19 16:47:01 +0000
@@ -119,13 +119,7 @@
 	}
 
 	if (icon_info != NULL) {
-#if GTK_CHECK_VERSION(3, 8, 0)
-		g_object_unref(icon_info);
-#else
-		/* NOTE: Leaving this in for lower version as it seems
-		   the object_unref() doesn't work on earlier versions. */
 		gtk_icon_info_free (icon_info);
-#endif
 	}
 }
 

