--- a/src/idomenuitemfactory.c
+++ b/src/idomenuitemfactory.c
@@ -18,7 +18,6 @@
  */
 
 #include <gtk/gtk.h>
-#include <gtk/ubuntu-private.h>
 
 #include "idoalarmmenuitem.h"
 #include "idoappointmentmenuitem.h"
@@ -32,23 +31,23 @@
 #include "idoapplicationmenuitem.h"
 #include "idosourcemenuitem.h"
 
-#define IDO_TYPE_MENU_ITEM_FACTORY         (ido_menu_item_factory_get_type ())
-#define IDO_MENU_ITEM_FACTORY(o)           (G_TYPE_CHECK_INSTANCE_CAST ((o), IDO_TYPE_MENU_ITEM_FACTORY, IdoMenuItemFactory))
-#define IDO_IS_MENU_ITEM_FACTORY(o)        (G_TYPE_CHECK_INSTANCE_TYPE ((o), IDO_TYPE_MENU_ITEM_FACTORY))
+//#define IDO_TYPE_MENU_ITEM_FACTORY         (ido_menu_item_factory_get_type ())
+//#define IDO_MENU_ITEM_FACTORY(o)           (G_TYPE_CHECK_INSTANCE_CAST ((o), IDO_TYPE_MENU_ITEM_FACTORY, IdoMenuItemFactory))
+//#define IDO_IS_MENU_ITEM_FACTORY(o)        (G_TYPE_CHECK_INSTANCE_TYPE ((o), IDO_TYPE_MENU_ITEM_FACTORY))
 
 typedef GObject      IdoMenuItemFactory;
 typedef GObjectClass IdoMenuItemFactoryClass;
 
-GType       ido_menu_item_factory_get_type (void);
-static void ido_menu_item_factory_interface_init (UbuntuMenuItemFactoryInterface *iface);
+//GType       ido_menu_item_factory_get_type (void);
+static void ido_menu_item_factory_interface_init (void *iface);
 
-G_DEFINE_TYPE_WITH_CODE (IdoMenuItemFactory, ido_menu_item_factory, G_TYPE_OBJECT,
-  G_IMPLEMENT_INTERFACE (UBUNTU_TYPE_MENU_ITEM_FACTORY, ido_menu_item_factory_interface_init)
-  g_io_extension_point_implement (UBUNTU_MENU_ITEM_FACTORY_EXTENSION_POINT_NAME,
-                                  g_define_type_id, "ido", 0);)
+//G_DEFINE_TYPE_WITH_CODE (IdoMenuItemFactory, ido_menu_item_factory, G_TYPE_OBJECT,
+//  G_IMPLEMENT_INTERFACE (UBUNTU_TYPE_MENU_ITEM_FACTORY, ido_menu_item_factory_interface_init)
+//  g_io_extension_point_implement (UBUNTU_MENU_ITEM_FACTORY_EXTENSION_POINT_NAME,
+//                                  g_define_type_id, "ido", 0);)
 
 static GtkMenuItem *
-ido_menu_item_factory_create_menu_item (UbuntuMenuItemFactory *factory,
+ido_menu_item_factory_create_menu_item (void                  *factory,
                                         const gchar           *type,
                                         GMenuItem             *menuitem,
                                         GActionGroup          *actions)
@@ -100,9 +99,9 @@
 }
 
 static void
-ido_menu_item_factory_interface_init (UbuntuMenuItemFactoryInterface *iface)
+ido_menu_item_factory_interface_init (void *iface)
 {
-  iface->create_menu_item = ido_menu_item_factory_create_menu_item;
+  //iface->create_menu_item = ido_menu_item_factory_create_menu_item;
 }
 
 static void
--- a/src/libido.c
+++ b/src/libido.c
@@ -28,9 +28,9 @@
 void
 ido_init (void)
 {
-  GType ido_menu_item_factory_get_type (void);
+//  GType ido_menu_item_factory_get_type (void);
 
   /* make sure this extension point is registered so that gtk calls it
    * when finding custom menu items */
-  g_type_ensure (ido_menu_item_factory_get_type ());
+//  g_type_ensure (ido_menu_item_factory_get_type ());
 }
