diff -Nru indicator-applet-0.5.0.orig/src/tomboykeybinder.h indicator-applet-0.5.0/src/tomboykeybinder.h
--- indicator-applet-0.5.0.orig/src/tomboykeybinder.h	2012-08-01 22:30:19.920847318 -0400
+++ indicator-applet-0.5.0/src/tomboykeybinder.h	2012-08-01 22:30:37.180753862 -0400
@@ -18,8 +18,6 @@
 #ifndef __TOMBOY_KEY_BINDER_H__
 #define __TOMBOY_KEY_BINDER_H__
 
-#include <glib/gtypes.h>
-
 G_BEGIN_DECLS
 
 typedef void (* TomboyBindkeyHandler) (char *keystring, gpointer user_data);
