Index: gnome-session-3.7.3/gnome-session/gsm-fail-whale.c
===================================================================
--- gnome-session-3.7.3.orig/gnome-session/gsm-fail-whale.c	2013-01-07 12:59:29.994981479 +1300
+++ gnome-session-3.7.3/gnome-session/gsm-fail-whale.c	2013-01-07 12:59:29.990981477 +1300
@@ -30,18 +30,10 @@
                                   gboolean            allow_logout,
                                   GsmShellExtensions *extensions)
 {
-        gint i;
-        gchar *argv[5];
+        static gboolean failed = FALSE;
+        if (failed)
+                return;
 
-        i = 0;
-        argv[i++] = LIBEXECDIR "/gnome-session-failed";
-        if (debug_mode)
-                argv[i++] = "--debug";
-        if (allow_logout)
-                argv[i++] = "--allow-logout";
-        if (extensions != NULL && gsm_shell_extensions_n_extensions (extensions) > 0)
-                argv[i++] = "--extensions";
-        argv[i++] = NULL;
-
-        g_spawn_async (NULL, argv, NULL, 0, NULL, NULL, NULL, NULL);
+        g_critical ("We failed, but the fail whale is dead. Sorry....");
+        failed = TRUE;
 }
