--- a/data/gnome.desktop.in
+++ b/data/gnome.desktop.in
@@ -1,7 +1,7 @@
 [Desktop Entry]
 _Name=GNOME
 _Comment=This session logs you into GNOME
-Exec=gnome-session
-TryExec=gnome-session
+Exec=gnome-session --session=gnome
+TryExec=gnome-shell
 Icon=
 Type=Application
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -13,7 +13,7 @@
 hwcompat_DATA = hardware-compatibility
 
 xsessiondir = $(datadir)/xsessions
-xsession_in_files = gnome.desktop.in
+xsession_in_files = gnome.desktop.in ubuntu.desktop.in
 
 if BUILD_SESSION_SELECTOR
 xsession_in_files += gnome-custom-session.desktop.in
@@ -26,7 +26,7 @@
 desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
 
 sessiondir = $(datadir)/gnome-session/sessions
-session_in_in_files = gnome.session.desktop.in.in gnome-dummy.session.desktop.in.in
+session_in_in_files = gnome.session.desktop.in.in gnome-dummy.session.desktop.in.in ubuntu.session.desktop.in.in
 session_in_files = $(session_in_in_files:.session.desktop.in.in=.session.desktop.in)
 session_DATA = $(session_in_files:.session.desktop.in=.session)
 
--- a/data/ubuntu.desktop.in
+++ b/data/ubuntu.desktop.in
@@ -0,0 +1,7 @@
+[Desktop Entry]
+_Name=Unity
+_Comment=This session logs you into Ubuntu
+Exec=gnome-session --session=ubuntu
+TryExec=unity
+Icon=
+Type=Application
--- a/data/ubuntu.session.desktop.in.in
+++ b/data/ubuntu.session.desktop.in.in
@@ -0,0 +1,3 @@
+[GNOME Session]
+_Name=Unity
+RequiredComponents=gnome-settings-daemon;compiz;
