--- a/data/gnome.session.desktop.in.in
+++ b/data/gnome.session.desktop.in.in
@@ -1,3 +1,4 @@
 [GNOME Session]
 _Name=GNOME
 RequiredComponents=gnome-shell;gnome-settings-daemon;
+DesktopName=GNOME
--- a/data/ubuntu.session.desktop.in.in
+++ b/data/ubuntu.session.desktop.in.in
@@ -1,3 +1,4 @@
 [GNOME Session] 
 _Name=Ubuntu
 RequiredComponents=gnome-settings-daemon;compiz;
+DesktopName=Unity
--- a/gnome-session/gsm-session-fill.c
+++ b/gnome-session/gsm-session-fill.c
@@ -31,6 +31,7 @@
 #define GSM_KEYFILE_SESSION_GROUP "GNOME Session"
 #define GSM_KEYFILE_RUNNABLE_KEY "IsRunnableHelper"
 #define GSM_KEYFILE_FALLBACK_KEY "FallbackSession"
+#define GSM_KEYFILE_DESKTOP_NAME_KEY "DesktopName"
 #define GSM_KEYFILE_REQUIRED_COMPONENTS_KEY "RequiredComponents"
 
 /* See https://bugzilla.gnome.org/show_bug.cgi?id=641992 for discussion */
@@ -314,6 +315,24 @@
         return keyfile;
 }
 
+static void
+set_xdg_current_desktop (GKeyFile *keyfile)
+{
+        char     *value;
+
+        value = g_key_file_get_string (keyfile,
+                                       GSM_KEYFILE_SESSION_GROUP, GSM_KEYFILE_DESKTOP_NAME_KEY,
+                                       NULL);
+
+        if (!IS_STRING_EMPTY (value)) {
+                gsm_util_setenv ("XDG_CURRENT_DESKTOP", value);
+        }
+        else {
+                gsm_util_setenv ("XDG_CURRENT_DESKTOP", "GNOME");
+        }
+        g_free (value);
+}
+
 gboolean
 gsm_session_fill (GsmManager  *manager,
                   const char  *session)
@@ -331,6 +350,8 @@
 
         g_free (actual_session);
 
+        set_xdg_current_desktop (keyfile);
+
         load_standard_apps (manager, keyfile);
 
         g_key_file_free (keyfile);
