diff -Nru gnome-session-3.4.1.orig/data/gnome-fallback.desktop.in gnome-session-3.4.1/data/gnome-fallback.desktop.in
--- gnome-session-3.4.1.orig/data/gnome-fallback.desktop.in	1969-12-31 19:00:00.000000000 -0500
+++ gnome-session-3.4.1/data/gnome-fallback.desktop.in	2012-04-24 14:30:22.540088275 -0400
@@ -0,0 +1,7 @@
+[Desktop Entry]
+_Name=GNOME fallback
+_Comment=This session logs you into GNOME
+Exec=gnome-session-fallback
+TryExec=gnome-session
+Icon=
+Type=Application
diff -Nru gnome-session-3.4.1.orig/data/Makefile.am gnome-session-3.4.1/data/Makefile.am
--- gnome-session-3.4.1.orig/data/Makefile.am	2012-04-24 14:27:37.270983615 -0400
+++ gnome-session-3.4.1/data/Makefile.am	2012-04-24 14:29:31.153699993 -0400
@@ -9,7 +9,7 @@
 hwcompat_DATA = hardware-compatibility
 
 xsessiondir = $(datadir)/xsessions
-xsession_in_files = gnome.desktop.in
+xsession_in_files = gnome.desktop.in gnome-fallback.desktop.in
 xsession_DATA = $(xsession_in_files:.desktop.in=.desktop)
 
 desktopdir = $(datadir)/applications
