diff -Nru gnome-session-3.4.1.orig/gnome-session/gsm-manager.c gnome-session-3.4.1/gnome-session/gsm-manager.c
--- gnome-session-3.4.1.orig/gnome-session/gsm-manager.c	2012-04-24 14:27:37.267650300 -0400
+++ gnome-session-3.4.1/gnome-session/gsm-manager.c	2012-04-24 14:34:04.892217024 -0400
@@ -504,7 +504,7 @@
                 g_signal_connect (system,
                                   "request-completed",
                                   G_CALLBACK (quit_request_completed),
-                                  GINT_TO_POINTER (GDM_LOGOUT_ACTION_REBOOT));
+                                  GINT_TO_POINTER (GDM_LOGOUT_ACTION_NONE));
                 gsm_system_attempt_restart (system);
                 break;
         case GSM_MANAGER_LOGOUT_REBOOT_GDM:
@@ -519,7 +519,7 @@
                 g_signal_connect (system,
                                   "request-completed",
                                   G_CALLBACK (quit_request_completed),
-                                  GINT_TO_POINTER (GDM_LOGOUT_ACTION_SHUTDOWN));
+                                  GINT_TO_POINTER (GDM_LOGOUT_ACTION_NONE));
                 gsm_system_attempt_stop (system);
                 break;
         case GSM_MANAGER_LOGOUT_SHUTDOWN_GDM:
