diff -Nru gnome-session-3.4.1.orig/data/gnome-classic.desktop.in gnome-session-3.4.1/data/gnome-classic.desktop.in
--- gnome-session-3.4.1.orig/data/gnome-classic.desktop.in	1969-12-31 19:00:00.000000000 -0500
+++ gnome-session-3.4.1/data/gnome-classic.desktop.in	2012-04-24 18:52:01.506810654 -0400
@@ -0,0 +1,7 @@
+[Desktop Entry]
+_Name=GNOME Classic
+_Comment=This session logs you into GNOME with the traditional panel
+Exec=gnome-session --session=gnome-classic
+TryExec=gnome-session
+Icon=
+Type=Application
diff -Nru gnome-session-3.4.1.orig/data/gnome-classic.session.desktop.in.in gnome-session-3.4.1/data/gnome-classic.session.desktop.in.in
--- gnome-session-3.4.1.orig/data/gnome-classic.session.desktop.in.in	1969-12-31 19:00:00.000000000 -0500
+++ gnome-session-3.4.1/data/gnome-classic.session.desktop.in.in	2012-04-24 18:52:24.333353658 -0400
@@ -0,0 +1,8 @@
+[GNOME Session]
+_Name=GNOME Classic
+RequiredComponents=gnome-panel;gnome-settings-daemon;
+RequiredProviders=windowmanager;
+DefaultProvider-windowmanager=gnome-wm
+DefaultProvider-notifications=notify-osd
+IsRunnableHelper=@LIBEXECDIR@/gnome-session-check-accelerated
+FallbackSession=gnome-fallback
diff -Nru gnome-session-3.4.1.orig/data/gnome.desktop.in gnome-session-3.4.1/data/gnome.desktop.in
--- gnome-session-3.4.1.orig/data/gnome.desktop.in	2012-04-24 14:27:37.270983615 -0400
+++ gnome-session-3.4.1/data/gnome.desktop.in	2012-04-24 18:49:24.737659945 -0400
@@ -1,7 +1,8 @@
 [Desktop Entry]
-_Name=GNOME
-_Comment=This session logs you into GNOME
-Exec=gnome-session
+_Name=Ubuntu
+_Comment=This session logs you into Ubuntu
+Exec=gnome-session --session=ubuntu
 TryExec=gnome-session
 Icon=
 Type=Application
+Hidden=true
diff -Nru gnome-session-3.4.1.orig/data/gnome-fallback.desktop.in gnome-session-3.4.1/data/gnome-fallback.desktop.in
--- gnome-session-3.4.1.orig/data/gnome-fallback.desktop.in	2012-04-24 14:30:58.909891244 -0400
+++ gnome-session-3.4.1/data/gnome-fallback.desktop.in	2012-04-24 18:54:26.886023067 -0400
@@ -1,7 +1,7 @@
 [Desktop Entry]
-_Name=GNOME fallback
-_Comment=This session logs you into GNOME
-Exec=gnome-session-fallback
+_Name=GNOME Classic (No effects)
+_Comment=This session logs you into GNOME with the traditional panel without any graphical effect.
+Exec=gnome-session --session=gnome-fallback
 TryExec=gnome-session
 Icon=
 Type=Application
diff -Nru gnome-session-3.4.1.orig/data/gnome-fallback.session.desktop.in.in gnome-session-3.4.1/data/gnome-fallback.session.desktop.in.in
--- gnome-session-3.4.1.orig/data/gnome-fallback.session.desktop.in.in	2012-04-24 14:28:45.380614634 -0400
+++ gnome-session-3.4.1/data/gnome-fallback.session.desktop.in.in	2012-04-24 18:53:27.603010897 -0400
@@ -1,6 +1,6 @@
 [GNOME Session]
-_Name=GNOME fallback
+_Name=GNOME fallback (Safe Mode)
 RequiredComponents=gnome-panel;gnome-settings-daemon;
-RequiredProviders=windowmanager;notifications;
-DefaultProvider-windowmanager=gnome-wm
-DefaultProvider-notifications=notification-daemon
+RequiredProviders=windowmanager;
+DefaultProvider-windowmanager=metacity
+DefaultProvider-notifications=notify-osd
diff -Nru gnome-session-3.4.1.orig/data/gnome-shell.desktop.in gnome-session-3.4.1/data/gnome-shell.desktop.in
--- gnome-session-3.4.1.orig/data/gnome-shell.desktop.in	1969-12-31 19:00:00.000000000 -0500
+++ gnome-session-3.4.1/data/gnome-shell.desktop.in	2012-04-24 18:48:11.301391117 -0400
@@ -0,0 +1,7 @@
+[Desktop Entry]
+_Name=GNOME
+_Comment=This session logs you into GNOME
+Exec=gnome-session --session=gnome
+TryExec=gnome-shell
+Icon=
+Type=Application
diff -Nru gnome-session-3.4.1.orig/data/Makefile.am gnome-session-3.4.1/data/Makefile.am
--- gnome-session-3.4.1.orig/data/Makefile.am	2012-04-24 14:30:58.909891244 -0400
+++ gnome-session-3.4.1/data/Makefile.am	2012-04-24 18:50:50.187197025 -0400
@@ -9,7 +9,7 @@
 hwcompat_DATA = hardware-compatibility
 
 xsessiondir = $(datadir)/xsessions
-xsession_in_files = gnome.desktop.in gnome-fallback.desktop.in
+xsession_in_files = gnome.desktop.in gnome-fallback.desktop.in gnome-shell.desktop.in gnome-classic.desktop.in ubuntu.desktop.in ubuntu-2d.desktop.in
 xsession_DATA = $(xsession_in_files:.desktop.in=.desktop)
 
 desktopdir = $(datadir)/applications
@@ -17,7 +17,7 @@
 desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
 
 sessiondir = $(datadir)/gnome-session/sessions
-session_in_in_files = gnome.session.desktop.in.in gnome-fallback.session.desktop.in.in
+session_in_in_files = gnome.session.desktop.in.in gnome-fallback.session.desktop.in.in gnome-classic.session.desktop.in.in ubuntu.session.desktop.in.in ubuntu-2d.session.desktop.in.in
 session_in_files = $(session_in_in_files:.session.desktop.in.in=.session.desktop.in)
 session_DATA = $(session_in_files:.session.desktop.in=.session)
 
diff -Nru gnome-session-3.4.1.orig/data/ubuntu-2d.desktop.in gnome-session-3.4.1/data/ubuntu-2d.desktop.in
--- gnome-session-3.4.1.orig/data/ubuntu-2d.desktop.in	1969-12-31 19:00:00.000000000 -0500
+++ gnome-session-3.4.1/data/ubuntu-2d.desktop.in	2012-04-24 18:51:20.737031523 -0400
@@ -0,0 +1,7 @@
+[Desktop Entry]
+_Name=Ubuntu 2D
+_Comment=This session logs you into Ubuntu 2D Mode
+Exec=gnome-session --session=ubuntu-2d
+TryExec=unity-2d-panel
+Icon=
+Type=Application
diff -Nru gnome-session-3.4.1.orig/data/ubuntu-2d.session.desktop.in.in gnome-session-3.4.1/data/ubuntu-2d.session.desktop.in.in
--- gnome-session-3.4.1.orig/data/ubuntu-2d.session.desktop.in.in	1969-12-31 19:00:00.000000000 -0500
+++ gnome-session-3.4.1/data/ubuntu-2d.session.desktop.in.in	2012-04-24 18:56:17.125425850 -0400
@@ -0,0 +1,8 @@
+[GNOME Session]
+_Name=Ubuntu 2D
+RequiredComponents=gnome-settings-daemon;
+RequiredProviders=windowmanager;panel;shell;
+DefaultProvider-windowmanager=metacity
+DefaultProvider-panel=unity-2d-panel
+DefaultProvider-shell=unity-2d-shell
+
diff -Nru gnome-session-3.4.1.orig/data/ubuntu.desktop.in gnome-session-3.4.1/data/ubuntu.desktop.in
--- gnome-session-3.4.1.orig/data/ubuntu.desktop.in	1969-12-31 19:00:00.000000000 -0500
+++ gnome-session-3.4.1/data/ubuntu.desktop.in	2012-04-24 18:56:48.008591873 -0400
@@ -0,0 +1,7 @@
+[Desktop Entry]
+_Name=Ubuntu
+_Comment=This session logs you into Ubuntu
+Exec=gnome-session --session=ubuntu
+TryExec=unity
+Icon=
+Type=Application
diff -Nru gnome-session-3.4.1.orig/data/ubuntu.session.desktop.in.in gnome-session-3.4.1/data/ubuntu.session.desktop.in.in
--- gnome-session-3.4.1.orig/data/ubuntu.session.desktop.in.in	1969-12-31 19:00:00.000000000 -0500
+++ gnome-session-3.4.1/data/ubuntu.session.desktop.in.in	2012-04-24 18:55:22.595721261 -0400
@@ -0,0 +1,8 @@
+[GNOME Session]
+_Name=Ubuntu
+RequiredComponents=gnome-settings-daemon;
+RequiredProviders=windowmanager;panel;
+DefaultProvider-windowmanager=compiz
+DefaultProvider-panel=compiz
+IsRunnableHelper=/usr/lib/nux/unity_support_test
+FallbackSession=ubuntu-2d
