diff -Nru gnome-session-3.4.1.orig/gnome-session/gsm-fail-whale-dialog.c gnome-session-3.4.1/gnome-session/gsm-fail-whale-dialog.c
--- gnome-session-3.4.1.orig/gnome-session/gsm-fail-whale-dialog.c	2012-04-24 14:27:37.267650300 -0400
+++ gnome-session-3.4.1/gnome-session/gsm-fail-whale-dialog.c	2012-04-24 19:32:51.276872443 -0400
@@ -355,26 +355,13 @@
                                  GsmShellExtensions *extensions)
 
 {
-        static GsmFailWhaleDialog *current_dialog = NULL;
-        GsmFailWhaleDialog        *fail_dialog;
+        static gboolean failed = FALSE;
 
-        if (current_dialog != NULL) {
+        if (failed) {
                 return;
         }
 
-        fail_dialog = g_object_new (GSM_TYPE_FAIL_WHALE_DIALOG, NULL);
-        fail_dialog->priv->debug_mode = debug_mode;
-        fail_dialog->priv->allow_logout = allow_logout;
-        fail_dialog->priv->extensions = extensions;
-
-        setup_window (fail_dialog);
-
-        current_dialog = fail_dialog;
-        g_signal_connect (current_dialog,
-                          "destroy",
-                          G_CALLBACK (gtk_widget_destroyed),
-                          &current_dialog);
-
-        gtk_widget_show (GTK_WIDGET (fail_dialog));
+        g_critical ("We failed, but the fail whale is dead. Sorry....");
+        failed = TRUE;
 }
 
