diff -Nru gnome-session-3.4.1.orig/data/session-properties.desktop.in.in gnome-session-3.4.1/data/session-properties.desktop.in.in
--- gnome-session-3.4.1.orig/data/session-properties.desktop.in.in	2012-04-24 14:27:37.270983615 -0400
+++ gnome-session-3.4.1/data/session-properties.desktop.in.in	2012-04-24 14:31:43.869647676 -0400
@@ -7,8 +7,7 @@
 Type=Application
 StartupNotify=true
 Categories=GTK;GNOME;Settings;X-GNOME-PersonalSettings;
-OnlyShowIn=GNOME;
-NoDisplay=true
+OnlyShowIn=GNOME;Unity;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-session
 X-GNOME-Bugzilla-Component=gnome-session-properties
