diff -Nru gnome-session-3.4.1.orig/gnome-session/gsm-manager.c gnome-session-3.4.1/gnome-session/gsm-manager.c
--- gnome-session-3.4.1.orig/gnome-session/gsm-manager.c	2012-04-24 19:23:59.106422124 -0400
+++ gnome-session-3.4.1/gnome-session/gsm-manager.c	2012-04-24 19:34:52.886213628 -0400
@@ -1654,7 +1654,6 @@
                 do_phase_startup (manager);
                 break;
         case GSM_MANAGER_PHASE_RUNNING:
-                possibly_show_fallback_dialog (manager);
                 g_signal_emit (manager, signals[SESSION_RUNNING], 0);
                 update_idle (manager);
                 break;
