diff -Nru libcompizconfig-0.9.5.94.orig/CMakeLists.txt libcompizconfig-0.9.5.94/CMakeLists.txt
--- libcompizconfig-0.9.5.94.orig/CMakeLists.txt	2012-04-20 15:12:19.084672253 -0400
+++ libcompizconfig-0.9.5.94/CMakeLists.txt	2012-04-20 15:19:00.462497802 -0400
@@ -4,14 +4,14 @@
 
 set (CMAKE_MODULE_PATH_ORIG ${CMAKE_MODULE_PATH})
 set (LIBCOMPIZCONFIG_CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
-set (CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${LIBCOMPIZCONFIGZ_CMAKE_MODULE_PATH})
+set (CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${LIBCOMPIZCONFIGZ_CMAKE_MODULE_PATH} /usr/share/compiz/cmake)
 
 include (CompizCommon)
 include (CompizPackage)
 include (CheckLibraryExists)
 include (CheckIncludeFile)
 
-set (CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_SOURCE_DIR}/cmake)
+set (CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_SOURCE_DIR}/cmake /usr/share/compiz/cmake)
 
 # libcompizconfig package version number
 # An odd micro number indicates in-progress development.
