diff -Nru unity_webapps-2.1.orig/src/libunity-webapps/libunity_webapps-0.2.pc.in unity_webapps-2.1/src/libunity-webapps/libunity_webapps-0.2.pc.in
--- unity_webapps-2.1.orig/src/libunity-webapps/libunity_webapps-0.2.pc.in	2012-09-01 16:42:52.859692353 -0400
+++ unity_webapps-2.1/src/libunity-webapps/libunity_webapps-0.2.pc.in	2012-09-01 16:43:06.012693326 -0400
@@ -9,5 +9,5 @@
 Description: Sample library created by Anjuta project wizard.
 Version: @VERSION@
 Requires: 
-Libs: -L${libdir} -lunity_webapps
+Libs: -L${libdir} -lunity-webapps
 Cflags: -I${includedir}
diff -Nru unity_webapps-2.1.orig/src/libunity-webapps-repository/libunity-webapps-repository.pc.in unity_webapps-2.1/src/libunity-webapps-repository/libunity-webapps-repository.pc.in
--- unity_webapps-2.1.orig/src/libunity-webapps-repository/libunity-webapps-repository.pc.in	2012-09-01 16:42:52.851692353 -0400
+++ unity_webapps-2.1/src/libunity-webapps-repository/libunity-webapps-repository.pc.in	2012-09-01 16:43:29.007695269 -0400
@@ -9,5 +9,5 @@
 Description: Unity Webapps Repository Access and Manipulation
 Version: @VERSION@
 Requires: 
-Libs: -L${libdir} -lunity_webapps
+Libs: -L${libdir} -lunity-webapps
 Cflags: -I${includedir}
