--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -40,7 +40,6 @@
                    libedataserver-1.2>=3.5
                    libcanberra>=0.12
                    libnotify>=0.7.6
-                   url-dispatcher-1>=1
                    properties-cpp>=0.0.1)
 include_directories (SYSTEM ${SERVICE_DEPS_INCLUDE_DIRS})
 
--- a/include/datetime/actions-live.h
+++ b/include/datetime/actions-live.h
@@ -50,7 +50,6 @@
 
 protected:
     virtual void execute_command(const std::string& command);
-    virtual void dispatch_url(const std::string& url);
 };
 
 } // namespace datetime
--- a/src/actions-live.cpp
+++ b/src/actions-live.cpp
@@ -19,8 +19,6 @@
 
 #include <datetime/actions-live.h>
 
-#include <url-dispatcher.h>
-
 #include <glib.h>
 
 namespace unity {
@@ -49,11 +47,6 @@
     }
 }
 
-void LiveActions::dispatch_url(const std::string& url)
-{
-    url_dispatch_send(url.c_str(), nullptr, nullptr);
-}
-
 /***
 ****
 ***/
@@ -81,12 +74,10 @@
 
 void LiveActions::open_phone_settings()
 {
-    dispatch_url("settings:///system/time-date");
 }
 
 void LiveActions::open_phone_clock_app()
 {
-    dispatch_url("appid://com.ubuntu.clock/clock/current-user-version");
 }
 
 void LiveActions::open_planner_at(const DateTime& dt)
@@ -102,8 +93,6 @@
         if(appt.uid != uid)
             continue;
 
-        if (!appt.url.empty())
-            dispatch_url(appt.url);
         break;
     }
 }
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -32,8 +32,6 @@
 #include <glib/gi18n.h> // bindtextdomain()
 #include <gio/gio.h>
 
-#include <url-dispatcher.h>
-
 #include <locale.h>
 #include <cstdlib> // exit()
 
@@ -74,7 +72,6 @@
                 url = a.url.c_str();
             else // alarm doesn't have a URl associated with it; use a fallback
                 url = "appid://com.ubuntu.clock/clock/current-user-version";
-            url_dispatch_send(url, nullptr, nullptr);
         };
         auto snap_dismiss = [](const Appointment&){};
         snap(appt, snap_show, snap_dismiss);
--- a/tests/test-live-actions.cpp
+++ b/tests/test-live-actions.cpp
@@ -35,7 +35,6 @@
     virtual ~MockLiveActions() {}
 
 protected:
-    void dispatch_url(const std::string& url) { last_url = url; }
     void execute_command(const std::string& cmd) { last_cmd = cmd; }
 };
 
