--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,7 +39,6 @@
                    libecal-1.2>=3.5
                    libedataserver-1.2>=3.5
                    libnotify>=0.7.6
-                   url-dispatcher-1>=1
                    json-glib-1.0>=0.16.2)
 include_directories (${SERVICE_INCLUDE_DIRS})
 
--- a/src/service.c
+++ b/src/service.c
@@ -24,7 +24,6 @@
 #include <gio/gio.h>
 #include <libnotify/notify.h>
 #include <json-glib/json-glib.h>
-#include <url-dispatcher.h>
 
 #include "dbus-shared.h"
 #include "service.h"
@@ -440,8 +439,6 @@
   g_debug ("dispatching url \"%s\" for appointment \"%s\", which begins at %s",
            appt->url, appt->summary, str);
   g_free (str);
-
-  url_dispatch_send (appt->url, NULL, NULL);
 }
 
 static void
@@ -455,7 +452,6 @@
     {
       const gchar * url = gurl;
       g_debug ("dispatching url '%s'", url);
-      url_dispatch_send (url, NULL, NULL);
     }
 }
 
@@ -1478,7 +1474,6 @@
                              GVariant      * param  G_GNUC_UNUSED,
                              gpointer        gself  G_GNUC_UNUSED)
 {
-  url_dispatch_send ("settings:///system/time-date", NULL, NULL);
 }
 
 static void
@@ -1515,7 +1510,6 @@
                           gpointer        gself  G_GNUC_UNUSED)
 {
   const char * url = "appid://com.ubuntu.clock/clock/current-user-version";
-  url_dispatch_send (url, NULL, NULL);
 }
 
 static void
