--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,7 +39,6 @@
                    libedataserver-1.2>=3.5
                    libcanberra>=0.12
                    libnotify>=0.7.6
-                   url-dispatcher-1>=1
                    properties-cpp>=0.0.1)
 include_directories (SYSTEM ${SERVICE_DEPS_INCLUDE_DIRS})
 
--- a/include/datetime/actions-live.h
+++ b/include/datetime/actions-live.h
@@ -54,7 +54,6 @@
 
 protected:
     virtual void execute_command(const std::string& command);
-    virtual void dispatch_url(const std::string& url);
 };
 
 } // namespace datetime
--- a/src/actions-live.cpp
+++ b/src/actions-live.cpp
@@ -19,8 +19,6 @@
 
 #include <datetime/actions-live.h>
 
-#include <url-dispatcher.h>
-
 #include <glib.h>
 
 namespace unity {
@@ -49,12 +47,6 @@
     }
 }
 
-void LiveActions::dispatch_url(const std::string& url)
-{
-    g_debug("Dispatching url '%s'", url.c_str());
-    url_dispatch_send(url.c_str(), nullptr, nullptr);
-}
-
 /***
 ****
 ***/
@@ -131,13 +123,11 @@
 
 void LiveActions::phone_open_alarm_app()
 {
-    dispatch_url("appid://com.ubuntu.clock/clock/current-user-version");
 }
 
 void LiveActions::phone_open_appointment(const Appointment& appt)
 {
-    if (!appt.url.empty())
-        dispatch_url(appt.url);
+    if (!appt.url.empty()) {}
     else if (appt.has_alarms)
         phone_open_alarm_app();
     else
@@ -146,13 +136,10 @@
 
 void LiveActions::phone_open_calendar_app(const DateTime&)
 {
-    // does calendar app have a mechanism for specifying dates?
-    dispatch_url("appid://com.ubuntu.calendar/calendar/current-user-version");
 }
 
 void LiveActions::phone_open_settings_app()
 {
-    dispatch_url("settings:///system/time-date");
 }
 
 /***
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -35,8 +35,6 @@
 #include <glib/gi18n.h> // bindtextdomain()
 #include <gio/gio.h>
 
-#include <url-dispatcher.h>
-
 #include <locale.h>
 #include <cstdlib> // exit()
 
@@ -89,7 +87,6 @@
                 url = a.url.c_str();
             else // alarm doesn't have a URl associated with it; use a fallback
                 url = "appid://com.ubuntu.clock/clock/current-user-version";
-            url_dispatch_send(url, nullptr, nullptr);
         };
         auto snap_dismiss = [](const Appointment&){};
         snap(appt, snap_show, snap_dismiss);
--- a/tests/test-live-actions.cpp
+++ b/tests/test-live-actions.cpp
@@ -35,7 +35,6 @@
     virtual ~MockLiveActions() {}
 
 protected:
-    void dispatch_url(const std::string& url) { last_url = url; }
     void execute_command(const std::string& cmd) { last_cmd = cmd; }
 };
 
