--- a/configure.ac
+++ c/configure.ac
@@ -77,7 +77,6 @@
                            libecal-1.2 >= $ECAL_REQUIRED_VERSION
                            libical >= $ICAL_REQUIRED_VERSION
                            libedataserver-1.2 >= EDS_REQUIRED_VERSION
-                           libedataserverui-3.0 >= EDS_REQUIRED_VERSION
                            cairo >= CAIRO_REQUIRED_VERSION
                            gdk-3.0 >= GDK_REQUIRED_VERSION)
 
