diff -Nru indicator-datetime-0.3.1.orig/src/datetime-service.c indicator-datetime-0.3.1/src/datetime-service.c
--- indicator-datetime-0.3.1.orig/src/datetime-service.c	2012-02-03 19:15:05.774554675 -0500
+++ indicator-datetime-0.3.1/src/datetime-service.c	2012-02-03 19:15:25.947888953 -0500
@@ -31,7 +31,7 @@
 #include <math.h>
 #include <gconf/gconf-client.h>
 
-#include <libdbusmenu-gtk3/menuitem.h>
+#include <libdbusmenu-gtk/menuitem.h>
 #include <libdbusmenu-glib/server.h>
 #include <libdbusmenu-glib/client.h>
 #include <libdbusmenu-glib/menuitem.h>
diff -Nru indicator-datetime-0.3.1.orig/src/indicator-datetime.c indicator-datetime-0.3.1/src/indicator-datetime.c
--- indicator-datetime-0.3.1.orig/src/indicator-datetime.c	2012-02-03 19:15:05.774554675 -0500
+++ indicator-datetime-0.3.1/src/indicator-datetime.c	2012-02-03 19:15:35.747889413 -0500
@@ -41,9 +41,9 @@
 #include <libindicator/indicator-service-manager.h>
 
 /* DBusMenu */
-#include <libdbusmenu-gtk3/menu.h>
+#include <libdbusmenu-gtk/menu.h>
 #include <libido/libido.h>
-#include <libdbusmenu-gtk3/menuitem.h>
+#include <libdbusmenu-gtk/menuitem.h>
 
 #include "utils.h"
 #include "dbus-shared.h"
