From cc3094d728910b3848763f231399e4788f08cef0 Mon Sep 17 00:00:00 2001
From: Xiao-Long Chen <chenxiaolong@cxl.epac.to>
Date: Mon, 14 Apr 2014 16:41:15 -0400
Subject: [PATCH 1/2] Remove overlay-scrollbar from GTK_MODULES

---
 shell/main.c | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/shell/main.c b/shell/main.c
index 3bf139e..e326b5f 100644
--- a/shell/main.c
+++ b/shell/main.c
@@ -31,9 +31,32 @@
 
 #include "cc-application.h"
 
+void
+remove_substring(char *src, const char *remove)
+{
+  while (src = strstr (src, remove)) {
+    memmove (src, src + strlen (remove), strlen (src + strlen(remove)) + 1);
+  }
+}
+
+void
+remove_overlay_scrollbars ()
+{
+  char *temp = getenv ("GTK_MODULES");
+  if (temp == NULL) {
+    return;
+  }
+
+  char *modules = strdup (temp);
+  remove_substring (modules, "overlay-scrollbar");
+  setenv ("GTK_MODULES", modules, 1);
+  free (modules);
+}
+
 int
 main (int argc, char **argv)
 {
+  remove_overlay_scrollbars ();
   GtkApplication *application;
   int status;
 
-- 
1.9.2

