diff -Nru gnome-control-center-3.5.92.orig/panels/keyboard/gnome-keyboard-panel.desktop.in.in gnome-control-center-3.5.92/panels/keyboard/gnome-keyboard-panel.desktop.in.in
--- gnome-control-center-3.5.92.orig/panels/keyboard/gnome-keyboard-panel.desktop.in.in	2012-09-21 19:29:50.592073021 -0400
+++ gnome-control-center-3.5.92/panels/keyboard/gnome-keyboard-panel.desktop.in.in	2012-09-21 19:36:03.654886992 -0400
@@ -2,7 +2,7 @@
 _Name=Keyboard
 _Comment=Change keyboard settings
 Exec=gnome-control-center keyboard
-Icon=input-keyboard
+Icon=preferences-desktop-keyboard
 Terminal=false
 Type=Application
 StartupNotify=true
