diff -Nru gnome-control-center-3.5.92.orig/panels/screen/cc-screen-panel.c gnome-control-center-3.5.92/panels/screen/cc-screen-panel.c
--- gnome-control-center-3.5.92.orig/panels/screen/cc-screen-panel.c	2012-09-21 19:32:05.389009874 -0400
+++ gnome-control-center-3.5.92/panels/screen/cc-screen-panel.c	2012-09-21 20:12:32.270341574 -0400
@@ -541,6 +541,13 @@
 
   update_lock_screen_sensitivity (self);
 
+  /* bind the screen lock suspend checkbutton */
+  widget = WID ("screen_lock_suspend_checkbutton");
+  g_settings_bind (self->priv->lock_settings,
+                   "ubuntu-lock-on-suspend",
+                   widget, "active",
+                   G_SETTINGS_BIND_DEFAULT);
+
   widget = WID ("screen_vbox");
   gtk_widget_reparent (widget, (GtkWidget *) self);
   g_object_set (self, "valign", GTK_ALIGN_START, NULL);
diff -Nru gnome-control-center-3.5.92.orig/panels/screen/screen.ui gnome-control-center-3.5.92/panels/screen/screen.ui
--- gnome-control-center-3.5.92.orig/panels/screen/screen.ui	2012-09-21 19:32:05.390009873 -0400
+++ gnome-control-center-3.5.92/panels/screen/screen.ui	2012-09-21 20:12:32.271341575 -0400
@@ -298,6 +298,31 @@
                           </packing>
                         </child>
                         <child>
+                          <object class="GtkHBox" id="screen_suspend_hbox">
+                            <property name="visible">True</property>
+                            <property name="can_focus">False</property>
+                            <property name="spacing">6</property>
+                            <child>
+                              <object class="GtkCheckButton" id="screen_lock_suspend_checkbutton">
+                                <property name="label" translatable="yes">Require my password when waking from suspend</property>
+                                <property name="use_action_appearance">False</property>
+                                <property name="visible">True</property>
+                                <property name="can_focus">True</property>
+                              </object>
+                              <packing>
+                                <property name="expand">False</property>
+                                <property name="fill">False</property>
+                                <property name="position">0</property>
+                              </packing>
+                            </child>
+                          </object>
+                          <packing>
+                            <property name="expand">False</property>
+                            <property name="fill">False</property>
+                            <property name="position">2</property>
+                          </packing>
+                        </child>
+                        <child>
                           <object class="GtkHBox" id="hbox2">
                             <property name="can_focus">False</property>
                             <property name="no_show_all">True</property>
@@ -339,7 +364,7 @@
                           <packing>
                             <property name="expand">True</property>
                             <property name="fill">True</property>
-                            <property name="position">2</property>
+                            <property name="position">3</property>
                           </packing>
                         </child>
                         <child>
