--- a/tests/a11y/Makefile.am
+++ b/tests/a11y/Makefile.am
@@ -44,17 +44,17 @@
 	hello-world.ui hello-world.txt \
 	mnemonic.ui mnemonic.txt \
 	accessible-name.ui accessible-name.txt \
-	notebook.ui notebook.txt \
+	notebook.txt \
 	range.ui range.txt \
 	link.ui link.txt \
 	text.ui text.txt \
 	buttons.ui buttons.txt \
 	colorchooser.ui colorchooser.txt \
-	about.ui about.txt \
+	about.txt \
 	messagedialog.ui messagedialog.txt \
 	expander.ui expander.txt \
 	assistant.ui assistant.txt \
-	pickers.ui pickers.txt \
+	pickers.txt \
 	lockbutton.ui lockbutton.txt \
 	spinner.ui spinner.txt \
 	progress.ui progress.txt \
@@ -66,5 +66,5 @@
 	entries.ui entries.txt \
 	scale-drawvalue.ui scale-drawvalue.txt \
 	placeholder-text.ui placeholder-text.txt \
-	menu.ui menu.txt \
-	combos.ui combos.txt
+	menu.txt \
+	combos.txt
--- a/tests/css/parser/Makefile.am
+++ b/tests/css/parser/Makefile.am
@@ -157,9 +157,7 @@
 	background-position-errors.ref.css \
 	background-repeat.css \
 	background-repeat.ref.css \
-	background-shorthand.css \
 	background-shorthand.ref.css \
-	background-shorthand-single.css \
 	background-shorthand-single.ref.css \
         background-size.css \
         background-size.ref.css \
@@ -262,9 +260,7 @@
 	declarations-valid-20.ref.css \
 	declarations-valid-21.css \
 	declarations-valid-21.ref.css \
-	declarations-valid-22.css \
 	declarations-valid-22.ref.css \
-	declarations-valid-23.css \
 	declarations-valid-23.ref.css \
 	declarations-valid-24.css \
 	declarations-valid-24.ref.css \
