diff -Nru gtk+-2.24.14.orig/configure.in gtk+-2.24.14/configure.in
--- gtk+-2.24.14.orig/configure.in	2013-01-09 07:05:59.818450979 -0500
+++ gtk+-2.24.14/configure.in	2013-01-09 07:08:03.701126480 -0500
@@ -51,7 +51,7 @@
 cflags_set=${CFLAGS+set}
 
 AM_INIT_AUTOMAKE([no-define -Wno-portability dist-bzip2])
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 # Support silent build rules, requires at least automake-1.11. Enable
 # by either passing --enable-silent-rules to configure or passing V=0
@@ -300,7 +300,7 @@
 
 # Checks for programs.
 AC_ISC_POSIX
-AM_PROG_CC_STDC
+AC_PROG_CC
 AM_PROG_CC_C_O
 AC_PROG_INSTALL
 AC_PROG_MAKE_SET
