From f92c599ba4609d2fc0a03b05e9472363d857b662 Mon Sep 17 00:00:00 2001
From: Xiao-Long Chen <chenxiaolong@cxl.epac.to>
Date: Tue, 28 Oct 2014 23:30:51 -0400
Subject: [PATCH 3/4] mpris2 Position property should be int64

---
 src/mpris2-interfaces.vala | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/mpris2-interfaces.vala b/src/mpris2-interfaces.vala
index cb68e84..aa95c83 100644
--- a/src/mpris2-interfaces.vala
+++ b/src/mpris2-interfaces.vala
@@ -36,7 +36,7 @@ public interface MprisRoot : Object {
 public interface MprisPlayer : Object {
   // properties
   public abstract HashTable<string, Variant?> Metadata{owned get; set;}
-  public abstract int32 Position{owned get; set;}
+  public abstract int64 Position{owned get; set;}
   public abstract string PlaybackStatus{owned get; set;}  
   // methods
   public abstract async void PlayPause() throws IOError;
-- 
2.1.2

