diff -Nru indicator-sound-0.8.0.0.orig/src/indicator-sound.c indicator-sound-0.8.0.0/src/indicator-sound.c
--- indicator-sound-0.8.0.0.orig/src/indicator-sound.c	2012-02-03 19:35:54.811279861 -0500
+++ indicator-sound-0.8.0.0/src/indicator-sound.c	2012-02-03 19:38:20.967953374 -0500
@@ -23,11 +23,7 @@
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
 #include <gdk/gdkkeysyms.h>
-#if GTK_CHECK_VERSION(3, 0, 0)
-#include <libdbusmenu-gtk3/menu.h>
-#else
 #include <libdbusmenu-gtk/menu.h>
-#endif
 #include <libido/idoscalemenuitem.h>
 
 #include <gio/gio.h>
diff -Nru indicator-sound-0.8.0.0.orig/src/metadata-widget.h indicator-sound-0.8.0.0/src/metadata-widget.h
--- indicator-sound-0.8.0.0.orig/src/metadata-widget.h	2012-02-03 19:35:54.811279861 -0500
+++ indicator-sound-0.8.0.0/src/metadata-widget.h	2012-02-03 19:37:42.354618232 -0500
@@ -20,11 +20,7 @@
 #define __METADATA_WIDGET_H__
 
 #include <gtk/gtk.h>
-#if GTK_CHECK_VERSION(3, 0, 0)
-#include <libdbusmenu-gtk3/menuitem.h>
-#else
 #include <libdbusmenu-gtk/menuitem.h>
-#endif
 
 G_BEGIN_DECLS
 
diff -Nru indicator-sound-0.8.0.0.orig/src/mute-widget.h indicator-sound-0.8.0.0/src/mute-widget.h
--- indicator-sound-0.8.0.0.orig/src/mute-widget.h	2012-02-03 19:35:54.811279861 -0500
+++ indicator-sound-0.8.0.0/src/mute-widget.h	2012-02-03 19:38:31.927953881 -0500
@@ -22,11 +22,7 @@
 #include <glib.h>
 #include <glib-object.h>
 #include <gtk/gtk.h>
-#if GTK_CHECK_VERSION(3, 0, 0)
-#include <libdbusmenu-gtk3/menuitem.h>
-#else
 #include <libdbusmenu-gtk/menuitem.h>
-#endif
 #include <libindicator/indicator-object.h>
 
 G_BEGIN_DECLS
diff -Nru indicator-sound-0.8.0.0.orig/src/transport-widget.h indicator-sound-0.8.0.0/src/transport-widget.h
--- indicator-sound-0.8.0.0.orig/src/transport-widget.h	2012-02-03 19:35:54.811279861 -0500
+++ indicator-sound-0.8.0.0/src/transport-widget.h	2012-02-03 19:37:56.031285542 -0500
@@ -20,11 +20,7 @@
 #define __TRANSPORT_WIDGET_H__
 
 #include <gtk/gtk.h>
-#if GTK_CHECK_VERSION(3, 0, 0)
-#include <libdbusmenu-gtk3/menuitem.h>
-#else
 #include <libdbusmenu-gtk/menuitem.h>
-#endif
 
 #include "common-defs.h"
 
diff -Nru indicator-sound-0.8.0.0.orig/src/voip-input-widget.h indicator-sound-0.8.0.0/src/voip-input-widget.h
--- indicator-sound-0.8.0.0.orig/src/voip-input-widget.h	2012-02-03 19:35:54.811279861 -0500
+++ indicator-sound-0.8.0.0/src/voip-input-widget.h	2012-02-03 19:38:09.457952836 -0500
@@ -22,11 +22,7 @@
 #include <glib.h>
 #include <glib-object.h>
 #include <gtk/gtk.h>
-#if GTK_CHECK_VERSION(3, 0, 0)
-#include <libdbusmenu-gtk3/menuitem.h>
-#else
 #include <libdbusmenu-gtk/menuitem.h>
-#endif
 
 G_BEGIN_DECLS
 
diff -Nru indicator-sound-0.8.0.0.orig/src/volume-widget.h indicator-sound-0.8.0.0/src/volume-widget.h
--- indicator-sound-0.8.0.0.orig/src/volume-widget.h	2012-02-03 19:35:54.814613196 -0500
+++ indicator-sound-0.8.0.0/src/volume-widget.h	2012-02-03 19:38:42.064621029 -0500
@@ -22,11 +22,7 @@
 #include <glib.h>
 #include <glib-object.h>
 #include <gtk/gtk.h>
-#if GTK_CHECK_VERSION(3, 0, 0)
-#include <libdbusmenu-gtk3/menuitem.h>
-#else
 #include <libdbusmenu-gtk/menuitem.h>
-#endif
 #include <libindicator/indicator-object.h>
 
 G_BEGIN_DECLS
