From 8f7b9afc350fffa126616e7761e46a1e80c0b4b4 Mon Sep 17 00:00:00 2001
From: Xiao-Long Chen <chenxiaolong@cxl.epac.to>
Date: Fri, 11 Apr 2014 15:46:47 -0400
Subject: [PATCH 2/6] Remove code depending on accountsservices patch

---
 lib/main.vala | 46 +---------------------------------------------
 1 file changed, 1 insertion(+), 45 deletions(-)

diff --git a/lib/main.vala b/lib/main.vala
index 1cb3896..030b51e 100644
--- a/lib/main.vala
+++ b/lib/main.vala
@@ -254,29 +254,7 @@ public class Indicator.Keyboard.Service : Object {
 				Act.User? user = manager.get_user ((!) greeter_user);
 
 				if (user != null && ((!) user).is_loaded) {
-					VariantIter outer;
-					VariantIter inner;
-
-					var sources = ((!) user).input_sources;
-					sources.get ("aa{ss}", out outer);
-
-					while (outer.next ("a{ss}", out inner)) {
-						unowned string key;
-						unowned string value;
-
-						while (inner.next ("{&s&s}", out key, out value)) {
-							if (key == "xkb") {
-								source = value;
-								break;
-							}
-						}
-
-						if (source != null) {
-							break;
-						}
-					}
-
-					if (source == null) {
+					if (true) {
 						var layouts = ((!) user).xkeyboard_layouts;
 
 						if (layouts.length <= 0) {
@@ -450,28 +428,6 @@ public class Indicator.Keyboard.Service : Object {
 			if (user.is_loaded) {
 				var done = false;
 
-				VariantIter outer;
-				VariantIter inner;
-
-				var sources = user.input_sources;
-				sources.get ("aa{ss}", out outer);
-
-				while (outer.next ("a{ss}", out inner)) {
-					unowned string key;
-					unowned string source;
-
-					while (inner.next ("{&s&s}", out key, out source)) {
-						if (key == "xkb") {
-							done = true;
-
-							if (!added.contains (source)) {
-								list.add (source);
-								added.add (source);
-							}
-						}
-					}
-				}
-
 				if (!done) {
 					var layouts = user.xkeyboard_layouts;
 					foreach (var layout in layouts) {
-- 
2.1.2

