diff -Nru indicator-appmenu-0.3.2.orig/scripts/menu-pusher.c indicator-appmenu-0.3.2/scripts/menu-pusher.c
--- indicator-appmenu-0.3.2.orig/scripts/menu-pusher.c	2012-02-03 20:49:07.174818979 -0500
+++ indicator-appmenu-0.3.2/scripts/menu-pusher.c	2012-02-03 20:49:27.361486592 -0500
@@ -23,11 +23,7 @@
 #include <gio/gio.h>
 #include <libdbusmenu-glib/menuitem.h>
 #include <libdbusmenu-glib/server.h>
-#if GTK_CHECK_VERSION(3, 0, 0)
-#include <libdbusmenu-gtk3/menuitem.h>
-#else
 #include <libdbusmenu-gtk/menuitem.h>
-#endif
 #include "../src/dbus-shared.h"
 
 int
diff -Nru indicator-appmenu-0.3.2.orig/src/window-menus.c indicator-appmenu-0.3.2/src/window-menus.c
--- indicator-appmenu-0.3.2.orig/src/window-menus.c	2012-02-03 20:49:07.171485646 -0500
+++ indicator-appmenu-0.3.2/src/window-menus.c	2012-02-03 20:49:51.468154389 -0500
@@ -25,11 +25,7 @@
 
 #include <gtk/gtk.h>
 
-#if GTK_CHECK_VERSION(3, 0, 0)
-#include <libdbusmenu-gtk3/menu.h>
-#else
 #include <libdbusmenu-gtk/menu.h>
-#endif
 #include <glib.h>
 #include <gio/gio.h>
 
