From 82469230b0a365fc20f128db08c824e13fcdc357 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Thu, 17 Oct 2013 09:14:01 +0200
Subject: [PATCH 1/2] power: Use up_client_get_critical_action()

To get the configured critical action.
---
 plugins/power/gsd-power-manager.c | 31 +++++++++++++------------------
 1 file changed, 13 insertions(+), 18 deletions(-)

diff --git a/plugins/power/gsd-power-manager.c b/plugins/power/gsd-power-manager.c
index 7e50349..1b25ebe 100644
--- a/plugins/power/gsd-power-manager.c
+++ b/plugins/power/gsd-power-manager.c
@@ -1241,23 +1241,18 @@ engine_ups_discharging (GsdPowerManager *manager, UpDevice *device)
 }
 
 static GsdPowerActionType
-manager_critical_action_get (GsdPowerManager *manager,
-                             gboolean         is_ups)
+manager_critical_action_get (GsdPowerManager *manager)
 {
         GsdPowerActionType policy;
+        char *action;
 
-        policy = g_settings_get_enum (manager->priv->settings, "critical-battery-action");
-        if (policy == GSD_POWER_ACTION_SUSPEND) {
-                if (is_ups == FALSE &&
-                    up_client_get_can_suspend (manager->priv->up_client))
-                        return policy;
-                return GSD_POWER_ACTION_SHUTDOWN;
-        } else if (policy == GSD_POWER_ACTION_HIBERNATE) {
-                if (up_client_get_can_hibernate (manager->priv->up_client))
-                        return policy;
-                return GSD_POWER_ACTION_SHUTDOWN;
-        }
-
+        action = up_client_get_critical_action (manager->priv->up_client);
+        /* We don't make the difference between HybridSleep and Hibernate */
+        if (g_strcmp0 (action, "PowerOff") == 0)
+                policy = GSD_POWER_ACTION_SHUTDOWN;
+        else
+                policy = GSD_POWER_ACTION_HIBERNATE;
+        g_free (action);
         return policy;
 }
 
@@ -1270,7 +1265,7 @@ manager_critical_action_do (GsdPowerManager *manager,
         /* stop playing the alert as it's too late to do anything now */
         play_loop_stop (&manager->priv->critical_alert_timeout_id);
 
-        action_type = manager_critical_action_get (manager, is_ups);
+        action_type = manager_critical_action_get (manager);
         do_power_action_type (manager, action_type);
 
         return FALSE;
@@ -1495,7 +1490,7 @@ engine_charge_critical (GsdPowerManager *manager, UpDevice *device)
                 }
 
                 /* we have to do different warnings depending on the policy */
-                policy = manager_critical_action_get (manager, FALSE);
+                policy = manager_critical_action_get (manager);
 
                 /* use different text for different actions */
                 if (policy == GSD_POWER_ACTION_NOTHING) {
@@ -1665,7 +1660,7 @@ engine_charge_action (GsdPowerManager *manager, UpDevice *device)
                 title = _("Laptop battery critically low");
 
                 /* we have to do different warnings depending on the policy */
-                policy = manager_critical_action_get (manager, FALSE);
+                policy = manager_critical_action_get (manager);
 
                 /* use different text for different actions */
                 if (policy == GSD_POWER_ACTION_NOTHING) {
@@ -1703,7 +1698,7 @@ engine_charge_action (GsdPowerManager *manager, UpDevice *device)
                 title = _("UPS critically low");
 
                 /* we have to do different warnings depending on the policy */
-                policy = manager_critical_action_get (manager, TRUE);
+                policy = manager_critical_action_get (manager);
 
                 /* use different text for different actions */
                 if (policy == GSD_POWER_ACTION_NOTHING) {
-- 
1.9.1

