From 343c52c5d648ecfefbc0734e0f7cdb40a626d532 Mon Sep 17 00:00:00 2001
From: Xiao-Long Chen <chenxiaolong@cxl.epac.to>
Date: Fri, 31 Oct 2014 15:07:27 -0400
Subject: [PATCH 1/4] Fix call to g_dbus_proxy_call_sync

---
 plugins/power/gsd-power-manager.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/plugins/power/gsd-power-manager.c b/plugins/power/gsd-power-manager.c
index c41e76c..eeb4425 100644
--- a/plugins/power/gsd-power-manager.c
+++ b/plugins/power/gsd-power-manager.c
@@ -1256,6 +1256,7 @@ manager_critical_action_get (GsdPowerManager *manager,
         } else if (policy == GSD_POWER_ACTION_HIBERNATE) {
                 result = g_dbus_proxy_call_sync (manager->priv->logind_proxy,
                                                  "CanHibernate",
+                                                 NULL,
                                                  G_DBUS_CALL_FLAGS_NONE,
                                                  -1, NULL, NULL);
         } else {
-- 
2.1.3

