diff -Nru unity-5.10.0.orig/UnityCore/GLibSignal-inl.h unity-5.10.0/UnityCore/GLibSignal-inl.h
--- unity-5.10.0.orig/UnityCore/GLibSignal-inl.h	2012-04-21 01:50:18.616933887 -0400
+++ unity-5.10.0/UnityCore/GLibSignal-inl.h	2012-04-21 01:52:35.372859684 -0400
@@ -228,7 +228,7 @@
 template<typename R, typename G, typename T1>
 Signal<R, G, T1>::Signal(G object, std::string const& signal_name, SignalCallback callback)
 {
-  Connect(object, signal_name, callback);
+  this->Connect(object, signal_name, callback);
 }
 
 template<typename R, typename G, typename T1, typename T2>
@@ -240,7 +240,7 @@
                              std::string const& signal_name,
                              SignalCallback callback)
 {
-  Connect(object, signal_name, callback);
+  this->Connect(object, signal_name, callback);
 }
 
 template<typename R, typename G, typename T1, typename T2, typename T3>
