--- a/cmake/CompizCommon.cmake
+++ b/cmake/CompizCommon.cmake
@@ -51,11 +51,6 @@
 # Almost everything is a shared library now, so almost everything needs -fPIC
 set (COMMON_FLAGS "-fPIC -Wall")
 
-option (COMPIZ_UNUSED_PRIVATE_FIELD_WARNINGS "Warn unused private fields" OFF)
-if (NOT COMPIZ_UNUSED_PRIVATE_FIELD_WARNINGS)
-    set (COMMON_FLAGS "${COMMON_FLAGS} -Wno-unused-private-field")
-endif ()
-
 option (COMPIZ_DEPRECATED_WARNINGS "Warn about declarations marked deprecated" OFF)
 if (NOT COMPIZ_DEPRECATED_WARNINGS)
     set (COMMON_FLAGS "${COMMON_FLAGS} -Wno-deprecated-declarations")
