From 649b9e0f6511ee6c9f4177d0d2efe8471ef0f6c6 Mon Sep 17 00:00:00 2001
From: Xiao-Long Chen <chenxiaolong@cxl.epac.to>
Date: Fri, 13 May 2016 23:30:44 -0400
Subject: [PATCH] Fix abs() compile error

---
 plugins/trailfocus/src/trailfocus.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/trailfocus/src/trailfocus.cpp b/plugins/trailfocus/src/trailfocus.cpp
index 1e20cfd..29fe60f 100644
--- a/plugins/trailfocus/src/trailfocus.cpp
+++ b/plugins/trailfocus/src/trailfocus.cpp
@@ -195,7 +195,7 @@ TrailfocusScreen::popWindow (TrailfocusWindow *tw)
 	}
 	else
 	{
-	    distance = abs (cur->activeNum () - best->activeNum ());
+	    distance = abs ((int) cur->activeNum () - (int) best->activeNum ());
 	    if (distance < bestDist)
 	    {
 		best     = cur;
-- 
2.8.2

