diff -Nru compiz-0.9.8+bzr3249.orig/cmake/base.cmake compiz-0.9.8+bzr3249/cmake/base.cmake
--- compiz-0.9.8+bzr3249.orig/cmake/base.cmake	2012-07-22 11:19:40.495649016 -0700
+++ compiz-0.9.8+bzr3249/cmake/base.cmake	2012-07-22 11:30:37.396046899 -0700
@@ -51,7 +51,7 @@
     install (FILES
 	     ${CMAKE_CURRENT_SOURCE_DIR}/cmake/FindCompiz.cmake
 	     DESTINATION
-	     ${CMAKE_INSTALL_PREFIX}/share/cmake-${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}/)
+	     ${CMAKE_INSTALL_PREFIX}/share/cmake-${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}/Modules/)
     add_custom_target (findcompiz_install
 	${CMAKE_COMMAND} -E make_directory ${CMAKE_ROOT}/Modules &&
 	${CMAKE_COMMAND} -E copy ${CMAKE_SOURCE_DIR}/cmake/FindCompiz.cmake ${CMAKE_ROOT}/Modules
@@ -67,7 +67,7 @@
     install (FILES
 	     ${CMAKE_CURRENT_SOURCE_DIR}/compizconfig/libcompizconfig/cmake/FindCompizConfig.cmake
 	     DESTINATION
-	     ${CMAKE_INSTALL_PREFIX}/share/cmake-${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}/)
+	     ${CMAKE_INSTALL_PREFIX}/share/cmake-${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}/Modules/)
     add_custom_target (
 	findcompizconfig_install
 	${CMAKE_COMMAND} -E make_directory ${CMAKE_ROOT}/Modules &&
