diff -Nru compiz-0.9.8.orig/compizconfig/cmake/exec_setup_py_with_destdir.cmake compiz-0.9.8/compizconfig/cmake/exec_setup_py_with_destdir.cmake
--- compiz-0.9.8.orig/compizconfig/cmake/exec_setup_py_with_destdir.cmake	2012-08-26 19:09:33.252597560 -0400
+++ compiz-0.9.8/compizconfig/cmake/exec_setup_py_with_destdir.cmake	2012-08-26 19:10:27.182598769 -0400
@@ -3,7 +3,6 @@
 set (SETUP "" CACHE FORCE "Path to setup.py")
 
 set (EXTRAARGS "")
-set (BUILD_DEB $ENV{COMPIZ_DEB_BUILD})
 
 set (INSTALL_ROOT $ENV{DESTDIR})
 
@@ -15,10 +14,5 @@
 
 cmake_policy (SET CMP0012 NEW)
 
-if (BUILD_DEB)
-    message ("Installing Debian Format")
-    set (EXTRAARGS "--install-layout=deb")
-endif (BUILD_DEB)
-
-execute_process (COMMAND python ${SETUP} install ${EXTRAARGS} --prefix=${PREFIX} ${INSTALL_ROOT_ARGS}
+execute_process (COMMAND python2 ${SETUP} install ${EXTRAARGS} --prefix=${PREFIX} ${INSTALL_ROOT_ARGS}
 	 		   WORKING_DIRECTORY ${WDIR})
