diff -Nru compiz-0.9.8.orig/compizconfig/ccsm/CMakeLists.txt compiz-0.9.8/compizconfig/ccsm/CMakeLists.txt
--- compiz-0.9.8.orig/compizconfig/ccsm/CMakeLists.txt	2012-08-26 19:09:33.252597561 -0400
+++ compiz-0.9.8/compizconfig/ccsm/CMakeLists.txt	2012-08-26 19:12:42.295935136 -0400
@@ -17,7 +17,7 @@
 execute_process (COMMAND uname -p OUTPUT_VARIABLE PROC_ARCH)
 
 # Get Python Version
-execute_process (COMMAND python -c "import sys; print str (sys.version_info.major) + '.' + str (sys.version_info.minor)"
+execute_process (COMMAND python2 -c "import sys; print str (sys.version_info.major) + '.' + str (sys.version_info.minor)"
 		 OUTPUT_VARIABLE PY_VERSION)
 
 string (REPLACE "\n" "" PROC_ARCH ${PROC_ARCH})
@@ -37,7 +37,7 @@
 		  ${CMAKE_CURRENT_BINARY_DIR}/build/${PY_SCRIPTS_DIR}/ccsm)
 
 add_custom_command (OUTPUT ${PY_CCSM_LIB}
-		    COMMAND python ${CMAKE_CURRENT_SOURCE_DIR}/setup.py build --build-base=${CMAKE_CURRENT_BINARY_DIR}/build
+		    COMMAND python2 ${CMAKE_CURRENT_SOURCE_DIR}/setup.py build --build-base=${CMAKE_CURRENT_BINARY_DIR}/build
 		    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
 		    COMMENT "Generating Python Bytecode")
 
@@ -46,7 +46,7 @@
 	     ${PY_CCSM_LIB})
 
 add_custom_target (uninstall
-		   COMMAND python ${CMAKE_CURRENT_SOURCE_DIR}/setup.py uninstall --prefix=${CMAKE_INSTALL_PREFIX}
+		   COMMAND python2 ${CMAKE_CURRENT_SOURCE_DIR}/setup.py uninstall --prefix=${CMAKE_INSTALL_PREFIX}
 		   WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
 		   COMMENT "Uninstalling Python Files")
 
diff -Nru compiz-0.9.8.orig/compizconfig/ccsm/Makefile compiz-0.9.8/compizconfig/ccsm/Makefile
--- compiz-0.9.8.orig/compizconfig/ccsm/Makefile	2012-08-26 19:09:33.252597560 -0400
+++ compiz-0.9.8/compizconfig/ccsm/Makefile	2012-08-26 19:12:56.942602130 -0400
@@ -1,11 +1,11 @@
 all:
-	@python setup.py build --prefix=${PREFIX}
+	@python2 setup.py build --prefix=${PREFIX}
 
 install: all
-	@python setup.py install --prefix=${PREFIX}
+	@python2 setup.py install --prefix=${PREFIX}
 
 uninstall:
-	@python setup.py uninstall --prefix=${PREFIX}
+	@python2 setup.py uninstall --prefix=${PREFIX}
 
 clean:
 	rm -rf build/
diff -Nru compiz-0.9.8.orig/compizconfig/ccsm/ccsm compiz-0.9.8/compizconfig/ccsm/ccsm
--- compiz-0.9.8.orig/compizconfig/ccsm/ccsm	2012-08-26 19:09:33.315930903 -0400
+++ compiz-0.9.8/compizconfig/ccsm/ccsm	2012-08-26 19:12:01.885934228 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: UTF-8 -*-
 
 # This program is free software; you can redistribute it and/or
diff -Nru compiz-0.9.8.orig/compizconfig/ccsm/setup.py compiz-0.9.8/compizconfig/ccsm/setup.py
--- compiz-0.9.8.orig/compizconfig/ccsm/setup.py	2012-08-26 19:09:33.322597565 -0400
+++ compiz-0.9.8/compizconfig/ccsm/setup.py	2012-08-26 19:13:12.452602480 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 import sys, os, glob
 import subprocess
diff -Nru compiz-0.9.8.orig/compizconfig/compizconfig-python/CMakeLists.txt compiz-0.9.8/compizconfig/compizconfig-python/CMakeLists.txt
--- compiz-0.9.8.orig/compizconfig/compizconfig-python/CMakeLists.txt	2012-08-26 19:09:33.232597556 -0400
+++ compiz-0.9.8/compizconfig/compizconfig-python/CMakeLists.txt	2012-08-26 19:13:52.159270037 -0400
@@ -17,7 +17,7 @@
 execute_process (COMMAND uname -p OUTPUT_VARIABLE PROC_ARCH)
 
 # Get Python Version
-execute_process (COMMAND python -c "import sys; print str (sys.version_info.major) + '.' + str (sys.version_info.minor)"
+execute_process (COMMAND python2 -c "import sys; print str (sys.version_info.major) + '.' + str (sys.version_info.minor)"
 		 OUTPUT_VARIABLE PY_VERSION)
 
 string (REPLACE "\n" "" PROC_ARCH ${PROC_ARCH})
@@ -39,7 +39,7 @@
 set (PY_COMPIZCONFIG_LIB  ${CMAKE_CURRENT_BINARY_DIR}/build/${PY_BUILD_DIR}/compizconfig.so)
 
 add_custom_command (OUTPUT ${PY_COMPIZCONFIG_LIB}
-		    COMMAND python ${CMAKE_CURRENT_SOURCE_DIR}/setup.py build --build-base=${CMAKE_CURRENT_BINARY_DIR}/build
+		    COMMAND python2 ${CMAKE_CURRENT_SOURCE_DIR}/setup.py build --build-base=${CMAKE_CURRENT_BINARY_DIR}/build
 		    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
 		    DEPENDS ${PY_COMPIZCONFIG_SRC} ${PY_COMPIZCONFIG_VERSION}
 		    COMMENT "Generating Python Bytecode")
@@ -49,7 +49,7 @@
 	     ${PY_COMPIZCONFIG_LIB})
 
 add_custom_target (uninstall
-		   COMMAND python ${CMAKE_CURRENT_SOURCE_DIR}/setup.py uninstall --prefix=${CMAKE_INSTALL_PREFIX}
+		   COMMAND python2 ${CMAKE_CURRENT_SOURCE_DIR}/setup.py uninstall --prefix=${CMAKE_INSTALL_PREFIX}
 		   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
 		   COMMENT "Uninstalling Python Files")
 
@@ -63,7 +63,7 @@
 if (COMPIZ_BUILD_TESTING)
 
         find_program (BASH_EXECUTABLE bash)
-	find_program (PYTHON_EXECUTABLE python)
+	find_program (PYTHON_EXECUTABLE python2)
 	mark_as_advanced (FORCE PYTHON_EXECUTABLE)
 
 	file (GLOB PYTHON_TESTS "tests/test_*.py")
diff -Nru compiz-0.9.8.orig/compizconfig/compizconfig-python/Makefile compiz-0.9.8/compizconfig/compizconfig-python/Makefile
--- compiz-0.9.8.orig/compizconfig/compizconfig-python/Makefile	2012-08-26 19:09:33.232597556 -0400
+++ compiz-0.9.8/compizconfig/compizconfig-python/Makefile	2012-08-26 19:14:16.299270574 -0400
@@ -1,11 +1,11 @@
 all:
-	@python setup.py build --prefix=${PREFIX}
+	@python2 setup.py build --prefix=${PREFIX}
 
 install: all
-	@python setup.py install --prefix=${PREFIX}
+	@python2 setup.py install --prefix=${PREFIX}
 
 uninstall:
-	@python setup.py uninstall --prefix=${PREFIX}
+	@python2 setup.py uninstall --prefix=${PREFIX}
 
 clean:
 	rm -rf build/
