diff -Nru compiz-0.9.9~daily12.12.05.orig/cmake/CompizCommon.cmake compiz-0.9.9~daily12.12.05/cmake/CompizCommon.cmake
--- compiz-0.9.9~daily12.12.05.orig/cmake/CompizCommon.cmake	2013-01-07 08:49:49.878610382 -0500
+++ compiz-0.9.9~daily12.12.05/cmake/CompizCommon.cmake	2013-01-07 08:50:54.194935065 -0500
@@ -53,7 +53,7 @@
 endif ()
 
 if (${CMAKE_PROJECT_NAME} STREQUAL "compiz")
-    set (COMPIZ_WERROR_DEFAULT ON)
+    set (COMPIZ_WERROR_DEFAULT OFF)
 else ()
     set (COMPIZ_WERROR_DEFAULT OFF)
 endif ()
