diff -Nru gnome-bluetooth-3.2.1.orig/applet/Makefile-lib.am gnome-bluetooth-3.2.1.fixed/applet/Makefile-lib.am
--- gnome-bluetooth-3.2.1.orig/applet/Makefile-lib.am	2012-01-04 17:59:31.791255903 -0500
+++ gnome-bluetooth-3.2.1.fixed/applet/Makefile-lib.am	2012-01-04 18:42:21.006772840 -0500
@@ -38,7 +38,8 @@
 GnomeBluetoothApplet_1_0_gir_FILES = $(libgnome_bluetooth_applet_la_SOURCES) ../lib/bluetooth-enums.h
 INTROSPECTION_GIRS += GnomeBluetoothApplet-1.0.gir
 
-pkglib_DATA = $(INTROSPECTION_GIRS:.gir=.typelib)
+typelibdir = $(pkglibdir)
+typelib_DATA = $(INTROSPECTION_GIRS:.gir=.typelib)
 
-CLEANFILES += $(pkglib_DATA)
+CLEANFILES += $(typelib_DATA)
 endif # HAVE_INTROSPECTION
