diff -Nru gnome-bluetooth-3.4.0.orig/applet/bluetooth-applet.desktop.in.in gnome-bluetooth-3.4.0/applet/bluetooth-applet.desktop.in.in
--- gnome-bluetooth-3.4.0.orig/applet/bluetooth-applet.desktop.in.in	2012-04-26 02:10:11.666047134 -0400
+++ gnome-bluetooth-3.4.0/applet/bluetooth-applet.desktop.in.in	2012-04-26 02:16:01.050821023 -0400
@@ -11,5 +11,5 @@
 X-GNOME-Bugzilla-Product=gnome-bluetooth
 X-GNOME-Bugzilla-Component=applet
 X-GNOME-Bugzilla-Version=@VERSION@
-AutostartCondition=GNOME3 if-session gnome-fallback
+AutostartCondition=GNOME3 unless-session gnome
 NoDisplay=true
