diff -Nru gnome-bluetooth-3.4.0.orig/applet/main.c gnome-bluetooth-3.4.0/applet/main.c
--- gnome-bluetooth-3.4.0.orig/applet/main.c	2012-04-26 02:03:49.658116651 -0400
+++ gnome-bluetooth-3.4.0/applet/main.c	2012-04-26 02:22:07.362169877 -0400
@@ -54,6 +54,7 @@
 #define KEYBOARD_PREFS		GNOMECC " keyboard"
 #define MOUSE_PREFS		GNOMECC " mouse"
 #define SOUND_PREFS		GNOMECC " sound"
+#define SOUND_NUA_PREFS		GNOMECC " sound-nua"
 
 enum {
 	CONNECTED,
@@ -144,7 +145,11 @@
 
 static void sound_callback(GObject *widget, gpointer user_data)
 {
-	const char *command = SOUND_PREFS;
+	char *command;
+	if (!g_strcmp0 (g_getenv ("XDG_CURRENT_DESKTOP"), "Unity"))
+		command = SOUND_NUA_PREFS;
+	else    
+		command = SOUND_PREFS;
 
 	if (!g_spawn_command_line_async(command, NULL))
 		g_printerr("Couldn't execute command: %s\n", command);
