From 4986dad712cef4e7d6b8444134ac41e70c6eb06b Mon Sep 17 00:00:00 2001
From: Xiao-Long Chen <chenxiaolong@cxl.epac.to>
Date: Sun, 8 Jun 2014 16:31:50 -0400
Subject: [PATCH 2/6] Skip bluetooth panel

---
 configure.ac | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index c91ee68..700f551 100644
--- a/configure.ac
+++ b/configure.ac
@@ -171,9 +171,7 @@ fi
 AM_CONDITIONAL(BUILD_NETWORK, [test x$have_networkmanager = xyes])
 
 # Check for gnome-bluetooth
-PKG_CHECK_MODULES(BLUETOOTH, $COMMON_MODULES gnome-bluetooth-1.0 >= 3.5.5,
-		  [have_bluetooth=yes], have_bluetooth=no)
-AM_CONDITIONAL(BUILD_BLUETOOTH, [test x$have_bluetooth = xyes])
+AM_CONDITIONAL(BUILD_BLUETOOTH, [false])
 
 # Check for CUPS 1.4 or newer
 AC_ARG_ENABLE([cups],
-- 
2.0.0

