From 7608f1b284dbabbf0ddc3461fb5d6dca5bb94890 Mon Sep 17 00:00:00 2001
From: Xiao-Long Chen <chenxiaolong@cxl.epac.to>
Date: Sat, 11 Oct 2014 18:50:22 -0400
Subject: [PATCH 3/8] Skip region panel

---
 panels/Makefile.am | 1 -
 1 file changed, 1 deletion(-)

diff --git a/panels/Makefile.am b/panels/Makefile.am
index b55b8d4..dc755c8 100644
--- a/panels/Makefile.am
+++ b/panels/Makefile.am
@@ -7,7 +7,6 @@ SUBDIRS= \
 	color \
 	display \
 	mouse \
-	region \
 	info \
 	sound \
 	keyboard \
-- 
2.1.2

