--- a/panels/common/cc-common-language.c
+++ b/panels/common/cc-common-language.c
@@ -326,66 +326,6 @@
         return language;
 }
 
-gchar *
-cc_common_language_get_property (const gchar *prop_name)
-{
-	GDBusConnection  *bus;
-	gchar            *user_path;
-	GError           *error = NULL;
-	GVariant         *properties;
-	GVariantIter     *iter;
-	gchar            *key;
-	GVariant         *value;
-	gchar            *ret = NULL;
-
-	if (g_strcmp0 (prop_name, "Language") != 0 && g_strcmp0 (prop_name, "FormatsLocale") != 0) {
-		g_warning ("Invalid argument: '%s'", prop_name);
-		return ret;
-	}
-
-	bus = g_bus_get_sync (G_BUS_TYPE_SYSTEM, NULL, NULL);
-	user_path = g_strdup_printf ("/org/freedesktop/Accounts/User%i", getuid ());
-
-	properties = g_dbus_connection_call_sync (bus,
-	                                          "org.freedesktop.Accounts",
-	                                          user_path,
-	                                          "org.freedesktop.DBus.Properties",
-	                                          "GetAll",
-	                                          g_variant_new ("(s)", "org.freedesktop.Accounts.User"),
-	                                          G_VARIANT_TYPE ("(a{sv})"),
-	                                          G_DBUS_CALL_FLAGS_NONE,
-	                                          -1,
-	                                          NULL,
-	                                          &error);
-	if (!properties) {
-		g_warning ("Error calling GetAll() when retrieving properties for %s: %s", user_path, error->message);
-		g_error_free (error);
-                /* g_hash_table_lookup() is not NULL-safe, so don't return NULL */
-                if (g_strcmp0 (prop_name, "Language") == 0)
-                        ret = g_strdup ("en");
-                else
-                        ret = g_strdup ("en_US.UTF-8");
-		goto out;
-	}
-
-	g_variant_get (properties, "(a{sv})", &iter);
-	while (g_variant_iter_loop (iter, "{&sv}", &key, &value)) {
-		if (g_strcmp0 (key, prop_name) == 0) {
-			g_variant_get (value, "s", &ret);
-			break;
-		}
-	}
-
-	g_variant_unref (properties);
-	g_variant_iter_free (iter);
-
-out:
-	g_object_unref (bus);
-	g_free (user_path);
-
-	return ret;
-}
-
 static void
 languages_foreach_cb (gpointer key,
 		      gpointer value,
@@ -467,7 +407,7 @@
 	char *lang;
 	gboolean found;
 
-	lang = cc_common_language_get_property ("Language");
+	lang = cc_common_language_get_current_language ();
 	g_debug ("Trying to select lang '%s' in treeview", lang);
 	model = gtk_tree_view_get_model (treeview);
 	found = FALSE;
@@ -522,7 +462,6 @@
         return ret;
 }
 
-/*
 static void
 add_other_users_language (GHashTable *ht)
 {
@@ -602,7 +541,6 @@
 
         g_object_unref (proxy);
 }
-*/
 
 GHashTable *
 cc_common_language_get_initial_languages (void)
@@ -614,7 +552,6 @@
         ht = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, g_free);
 
         /* Add some common languages first */
-/*
         g_hash_table_insert (ht, g_strdup ("en_US.utf8"), g_strdup (_("English")));
         if (gdm_language_has_translations ("en_GB"))
                 g_hash_table_insert (ht, g_strdup ("en_GB.utf8"), g_strdup (_("British English")));
@@ -635,29 +572,11 @@
         if (gdm_language_has_translations ("ar") ||
             gdm_language_has_translations ("ar_EG"))
                 g_hash_table_insert (ht, g_strdup ("ar_EG.utf8"), g_strdup (_("Arabic")));
-*/
         /* Add the languages used by other users on the system */
-//        add_other_users_language (ht);
-
-        /* Add installed languages */
-        gchar  *avail_languages;
-        GError *error = NULL;
-        if (g_spawn_command_line_sync ("/usr/share/language-tools/language-options",
-                                        &avail_languages, NULL, NULL, &error)) {
-                name = strtok (avail_languages, "\n");
-                while (name != NULL) {
-                        language = gdm_get_language_from_name (name, NULL);
-                        g_hash_table_insert (ht, g_strdup (name), language);
-                        name = strtok (NULL, "\n");
-                }
-                g_free (avail_languages);
-        } else {
-                g_warning ("Couldn't get available languages: %s", error->message);
-                g_error_free (error);
-        }
+        add_other_users_language (ht);
 
-        /* Add current language */
-        name = cc_common_language_get_property ("Language");
+        /* Add current locale */
+        name = cc_common_language_get_current_language ();
         if (g_hash_table_lookup (ht, name) == NULL) {
                 language = gdm_get_language_from_name (name, NULL);
                 g_hash_table_insert (ht, name, language);
--- a/panels/common/cc-common-language.h
+++ b/panels/common/cc-common-language.h
@@ -45,7 +45,6 @@
                                                      GHashTable       *user_langs);
 gboolean cc_common_language_has_font                (const gchar  *locale);
 gchar   *cc_common_language_get_current_language    (void);
-gchar   *cc_common_language_get_property            (const gchar  *prop_name);
 
 GHashTable *cc_common_language_get_initial_languages   (void);
 GHashTable *cc_common_language_get_initial_regions     (const gchar *lang);
--- a/panels/common/cc-language-chooser.c
+++ b/panels/common/cc-language-chooser.c
@@ -105,8 +105,8 @@
 
 	user_langs = cc_common_language_get_initial_languages ();
 
-	/* Add the current language first */
-	name = cc_common_language_get_property ("Language");
+	/* Add the current locale first */
+	name = cc_common_language_get_current_language ();
 	display = g_hash_table_lookup (user_langs, name);
 
         gtk_list_store_append (store, &iter);
@@ -118,8 +118,8 @@
 	g_hash_table_foreach (user_langs, (GHFunc) languages_foreach_cb, store);
 
 	/* And now the "Other..." selection */
-//        gtk_list_store_append (store, &iter);
-//        gtk_list_store_set (store, &iter, LOCALE_COL, NULL, DISPLAY_LOCALE_COL, _("Other..."), -1);
+        gtk_list_store_append (store, &iter);
+        gtk_list_store_set (store, &iter, LOCALE_COL, NULL, DISPLAY_LOCALE_COL, _("Other..."), -1);
 
         g_hash_table_destroy (user_langs);
 }
--- a/panels/common/gdm-languages.c
+++ b/panels/common/gdm-languages.c
@@ -224,7 +224,6 @@
                          const char *codeset,
                          const char *modifier)
 {
-        const char *adj_codeset;
         char *name;
 
         g_assert (language[0] != 0);
@@ -232,17 +231,12 @@
         g_assert (codeset == NULL || codeset[0] != 0);
         g_assert (modifier == NULL || modifier[0] != 0);
 
-        if (g_strcmp0 (codeset, "utf8") == 0) {
-                adj_codeset = "UTF-8";
-        } else
-                adj_codeset = codeset;
-
         name = g_strdup_printf ("%s%s%s%s%s%s%s",
                                 language,
                                 territory != NULL? "_" : "",
                                 territory != NULL? territory : "",
                                 codeset != NULL? "." : "",
-                                codeset != NULL? adj_codeset : "",
+                                codeset != NULL? codeset : "",
                                 modifier != NULL? "@" : "",
                                 modifier != NULL? modifier : "");
 
@@ -860,7 +854,6 @@
         const char *ccode_longT;
         const char *ccode;
         const char *ccode_id;
-        const char *lang_common_name;
         const char *lang_name;
 
         if (! (g_str_equal (element_name, "iso_639_entry") || g_str_equal (element_name, "iso_639_3_entry"))
@@ -872,7 +865,6 @@
         ccode_longB = NULL;
         ccode_longT = NULL;
         ccode_id = NULL;
-        lang_common_name = NULL;
         lang_name = NULL;
 
         while (*attr_names && *attr_values) {
@@ -909,11 +901,6 @@
                                 }
                                 ccode_id = *attr_values;
                         }
-                } else if (g_str_equal (*attr_names, "common_name")) {
-                        /* skip if empty */
-                        if (**attr_values) {
-                                lang_common_name = *attr_values;
-                        }
                 } else if (g_str_equal (*attr_names, "name")) {
                         lang_name = *attr_values;
                 }
@@ -922,10 +909,6 @@
                 ++attr_values;
         }
 
-        if (lang_common_name != NULL) {
-                lang_name = lang_common_name;
-        }
-
         if (lang_name == NULL) {
                 return;
         }
@@ -1149,7 +1132,6 @@
         char *langinfo_codeset;
         char *translated_language;
         char *translated_territory;
-        char *modifier;
         gboolean is_utf8 = TRUE;
 
         g_return_val_if_fail (name != NULL, NULL);
@@ -1172,13 +1154,12 @@
         language_code = NULL;
         territory_code = NULL;
         codeset_code = NULL;
-        modifier = NULL;
 
         gdm_parse_language_name (name,
                                  &language_code,
                                  &territory_code,
                                  &codeset_code,
-                                 &modifier);
+                                 NULL);
 
         if (language_code == NULL) {
                 goto out;
@@ -1204,7 +1185,7 @@
                                         translated_territory);
         }
 
-//        language_name_get_codeset_details (name, &langinfo_codeset, &is_utf8);
+        language_name_get_codeset_details (name, &langinfo_codeset, &is_utf8);
 
         if (codeset_code == NULL && langinfo_codeset != NULL) {
             codeset_code = g_strdup (langinfo_codeset);
@@ -1216,10 +1197,6 @@
                                         codeset_code);
         }
 
-        if (modifier != NULL) {
-                g_string_append_printf (full_language, " - %s", modifier);
-        }
-
 out:
        g_free (language_code);
        g_free (territory_code);
@@ -1227,7 +1204,6 @@
        g_free (langinfo_codeset);
        g_free (translated_language);
        g_free (translated_territory);
-       g_free (modifier);
 
        if (full_language->len == 0) {
                g_string_free (full_language, TRUE);
--- a/panels/region/gnome-region-panel-formats.c
+++ b/panels/region/gnome-region-panel-formats.c
@@ -31,7 +31,6 @@
 #include "cc-language-chooser.h"
 #include "gdm-languages.h"
 #include "gnome-region-panel-formats.h"
-#include "gnome-region-panel-system.h"
 
 static void
 display_date (GtkLabel *label, GDateTime *dt, const gchar *format)
@@ -151,47 +150,6 @@
 }
 
 static void
-set_formats_locale (const gchar *formats_locale)
-{
-	GDBusProxy  *proxy;
-	GError      *error = NULL;
-	gchar       *user_path;
-	GVariant    *ret;
-
-	user_path = g_strdup_printf ("/org/freedesktop/Accounts/User%i", getuid ());
-	proxy = g_dbus_proxy_new_for_bus_sync (G_BUS_TYPE_SYSTEM,
-	                                       G_DBUS_PROXY_FLAGS_NONE,
-	                                       NULL,
-	                                       "org.freedesktop.Accounts",
-	                                       user_path,
-	                                       "org.freedesktop.Accounts.User",
-	                                       NULL,
-	                                       &error);
-	if (!proxy) {
-		g_warning ("Couldn't get accountsservice proxy for %s: %s", user_path, error->message);
-		g_error_free (error);
-		g_free (user_path);
-		return;
-	}
-
-	ret = g_dbus_proxy_call_sync (proxy,
-	                              "SetFormatsLocale",
-	                              g_variant_new ("(s)", formats_locale),
-	                              G_DBUS_CALL_FLAGS_NONE,
-	                              -1,
-	                              NULL,
-	                              &error);
-	if (!ret) {
-		g_warning ("Couldn't set FormatsLocale: %s", error->message);
-		g_error_free (error);
-	} else
-		g_variant_unref (ret);
-
-	g_object_unref (proxy);
-	g_free (user_path);
-}
-
-static void
 update_settings_cb (GtkTreeSelection *selection, gpointer user_data)
 {
         GtkBuilder *builder = GTK_BUILDER (user_data);
@@ -200,6 +158,7 @@
         gchar *active_id;
         GtkWidget *treeview;
         GSettings *locale_settings;
+        gchar *current_setting;
 
         if (!gtk_tree_selection_get_selected (selection, &model, &iter)) {
                 return;
@@ -209,10 +168,13 @@
         treeview = GTK_WIDGET (gtk_builder_get_object (builder, "region_selector"));
 
         locale_settings = g_object_get_data (G_OBJECT (treeview), "settings");
+        current_setting = g_settings_get_string (locale_settings, "region");
 
-        set_formats_locale (active_id);
-        locale_settings_changed (locale_settings, NULL, builder);
+        if (g_strcmp0 (active_id, current_setting) != 0) {
+                g_settings_set_string (locale_settings, "region", active_id);
+        }
 
+        g_free (current_setting);
         g_free (active_id);
 }
 
@@ -221,7 +183,7 @@
 {
         gchar *current_setting;
 
-        current_setting = cc_common_language_get_property ("FormatsLocale");
+        current_setting = g_settings_get_string (locale_settings, "region");
         select_region (treeview, current_setting);
         g_free (current_setting);
 }
@@ -250,6 +212,7 @@
 populate_regions (GtkBuilder *builder, const gchar *current_lang)
 {
         gchar *current_region;
+        GSettings *locale_settings;
         GHashTable *ht;
         GHashTableIter htiter;
         GtkTreeModel *model;
@@ -264,10 +227,11 @@
         g_signal_handlers_block_by_func (selection, update_settings_cb, builder);
 
         model = gtk_tree_view_get_model (GTK_TREE_VIEW (treeview));
+        locale_settings = g_object_get_data (G_OBJECT (treeview), "settings");
 
         ht = cc_common_language_get_initial_regions (current_lang);
 
-        current_region = cc_common_language_get_property ("FormatsLocale");
+        current_region = g_settings_get_string (locale_settings, "region");
         if (!current_region || !current_region[0]) {
                 current_region = g_strdup (current_lang);
         }
--- a/panels/region/gnome-region-panel-system.c
+++ b/panels/region/gnome-region-panel-system.c
@@ -92,7 +92,7 @@
                 gtk_widget_set_sensitive (button, TRUE);
 }
 
-void
+static void
 locale_settings_changed (GSettings *settings,
                          const gchar *key,
                          GtkBuilder *dialog)
@@ -100,7 +100,7 @@
         GtkWidget *label;
         gchar *region, *display_region;
 
-        region = cc_common_language_get_property ("FormatsLocale");
+        region = g_settings_get_string (locale_settings, "region");
         if (!region || !region[0]) {
                 label = WID ("user_display_language");
                 region = g_strdup ((gchar*)g_object_get_data (G_OBJECT (label), "language"));
@@ -226,16 +226,6 @@
         }
 }
 
-static gchar *
-strip_quotes (const gchar *str)
-{
-        if ((g_str_has_prefix (str, "\"") && g_str_has_suffix (str, "\""))
-          || (g_str_has_prefix (str, "'") && g_str_has_suffix (str, "'")))
-                return g_strndup (str + 1, strlen (str) - 2);
-        else
-                return g_strdup (str);
-}
-
 static void
 on_localed_properties_changed (GDBusProxy   *proxy,
                                GVariant     *changed_properties,
@@ -267,43 +257,33 @@
                 const gchar **strv;
                 gsize len;
                 gint i;
-                gchar *lang, *language, *messages, *time;
+                const gchar *lang, *messages, *time;
                 gchar *name;
                 GtkWidget *label;
 
                 strv = g_variant_get_strv (v, &len);
 
-                lang = language = messages = time = NULL;
+                lang = messages = time = NULL;
                 for (i = 0; strv[i]; i++) {
                         if (g_str_has_prefix (strv[i], "LANG=")) {
-                                lang = strip_quotes (strv[i] + strlen ("LANG="));
-                        }
-                        else if (g_str_has_prefix (strv[i], "LANGUAGE=")) {
-                                gchar *tmp = strip_quotes (strv[i] + strlen ("LANGUAGE="));
-                                gchar **tokens = g_strsplit (tmp, ":", 2);
-                                language = g_strdup (tokens[0]);
-                                g_free (tmp);
-                                g_strfreev (tokens);
+                                lang = strv[i] + strlen ("LANG=");
                         }
                         else if (g_str_has_prefix (strv[i], "LC_MESSAGES=")) {
-                                messages = strip_quotes (strv[i] + strlen ("LC_MESSAGES="));
+                                messages = strv[i] + strlen ("LC_MESSAGES=");
                         }
                         else if (g_str_has_prefix (strv[i], "LC_TIME=")) {
-                                time = strip_quotes (strv[i] + strlen ("LC_TIME="));
+                                time = strv[i] + strlen ("LC_TIME=");
                         }
                 }
-                if (!language) {
-                        if (messages)
-                                language = g_strdup (messages);
-                        else
-                                language = g_strdup (lang);
+                if (!messages) {
+                        messages = lang;
                 }
                 if (!time) {
-                        time = g_strdup (lang);
+                        time = lang;
                 }
 
-                if (language) {
-                        name = gdm_get_language_from_name (language, NULL);
+                if (messages) {
+                        name = gdm_get_language_from_name (messages, NULL);
                         label = WID ("system_display_language");
                         gtk_label_set_text (GTK_LABEL (label), name);
                         g_free (name);
@@ -318,10 +298,6 @@
                         g_object_set_data_full (G_OBJECT (label), "region", g_strdup (time), g_free);
                 }
                 g_variant_unref (v);
-                g_free (lang);
-                g_free (language);
-                g_free (messages);
-                g_free (time);
         }
 
         label = WID ("system_input_source");
@@ -420,44 +396,27 @@
         GtkWidget *label;
         GVariantBuilder *b;
         gchar *s;
-        gchar *command;
-        gchar *lang;
-        GError *error = NULL;
-        gint i;
 
         label = WID ("user_display_language");
         language = g_object_get_data (G_OBJECT (label), "language");
         label = WID ("user_format");
         region = g_object_get_data (G_OBJECT (label), "region");
 
-        /* Get locale that corresponds to the language */
-        command = g_strconcat ("/usr/share/language-tools/language2locale ", language, NULL);
-        if (!g_spawn_command_line_sync (command, &lang, NULL, NULL, &error)) {
-                g_warning ("Couldn't get LANG locale: %s", error->message);
-                g_error_free (error);
-                g_free (command);
-                return;
-        }
-        g_free (command);
-        g_strchomp (lang);
-        if (strlen (lang) == 0) {
-                g_warning ("Couldn't get LANG locale -- Copying interrupted");
-                return;
-        }
-
         b = g_variant_builder_new (G_VARIANT_TYPE ("as"));
-        s = g_strconcat ("LANG=", lang, NULL);
+        s = g_strconcat ("LANG=", language, NULL);
         g_variant_builder_add (b, "s", s);
-        g_free (lang);
         g_free (s);
-        s = g_strconcat ("LANGUAGE=", language, NULL);
-        g_variant_builder_add (b, "s", s);
-        g_free (s);
-        const gchar *format_categories[] = { "LC_NUMERIC", "LC_TIME",
-           "LC_MONETARY", "LC_PAPER", "LC_IDENTIFICATION", "LC_NAME",
-           "LC_ADDRESS", "LC_TELEPHONE", "LC_MEASUREMENT", NULL };
-        for (i = 0; format_categories[i] != NULL; i++) {
-                s = g_strconcat (format_categories[i], "=", region, NULL);
+        if (g_strcmp0 (language, region) != 0) {
+                s = g_strconcat ("LC_TIME=", region, NULL);
+                g_variant_builder_add (b, "s", s);
+                g_free (s);
+                s = g_strconcat ("LC_NUMERIC=", region, NULL);
+                g_variant_builder_add (b, "s", s);
+                g_free (s);
+                s = g_strconcat ("LC_MONETARY=", region, NULL);
+                g_variant_builder_add (b, "s", s);
+                g_free (s);
+                s = g_strconcat ("LC_MEASUREMENT=", region, NULL);
                 g_variant_builder_add (b, "s", s);
                 g_free (s);
         }
@@ -566,7 +525,7 @@
         g_object_weak_ref (G_OBJECT (dialog), (GWeakNotify) g_object_unref, input_sources_settings);
 
         /* Display user settings */
-        language = cc_common_language_get_property ("Language");
+        language = cc_common_language_get_current_language ();
         system_update_language (dialog, language);
         g_free (language);
 
--- a/panels/region/gnome-region-panel-system.h
+++ b/panels/region/gnome-region-panel-system.h
@@ -27,8 +27,5 @@
 void setup_system           (GtkBuilder *builder);
 void system_update_language (GtkBuilder  *builder,
                              const gchar *language);
-void locale_settings_changed (GSettings   *settings,
-                              const gchar *key,
-                              GtkBuilder  *dialog);
 
 #endif
--- a/panels/region/gnome-region-panel.ui
+++ b/panels/region/gnome-region-panel.ui
@@ -154,7 +154,7 @@
                         </child>
                         <child>
                           <object class="GtkToolbar" id="language-toolbar">
-                            <property name="visible">False</property>
+                            <property name="visible">True</property>
                             <property name="can_focus">False</property>
                             <property name="toolbar_style">icons</property>
                             <property name="show_arrow">False</property>
