--- a/panels/color/cc-color-panel.c
+++ b/panels/color/cc-color-panel.c
@@ -2390,10 +2390,7 @@
 static const char *
 cc_color_panel_get_help_uri (CcPanel *panel)
 {
-  if (!g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "Unity"))
-    return "help:ubuntu-help/color";
-  else
-    return "help:gnome-help/color";
+  return "help:gnome-help/color";
 }
 
 static void
@@ -2675,12 +2672,6 @@
   g_signal_connect (widget, "realize",
                     G_CALLBACK (gcm_prefs_window_realize_cb),
                     prefs);
-
-  widget = WID (priv->builder, "linkbutton_help");
-  if (!g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "Unity"))
-    g_object_set (G_OBJECT (widget),
-              "uri", "help:ubuntu-help/color-whyimportant",
-              NULL);
 }
 
 void
--- a/panels/datetime/datetime-prefs.c
+++ b/panels/datetime/datetime-prefs.c
@@ -826,7 +826,7 @@
 static const char *
 indicator_datetime_panel_get_help_uri (CcPanel *panel G_GNUC_UNUSED)
 {
-  return "help:ubuntu-help/clock";
+  return "help:gnome-help/clock";
 }
 
 static void
--- a/panels/display/cc-display-panel.c
+++ b/panels/display/cc-display-panel.c
@@ -217,10 +217,7 @@
 static const char *
 cc_display_panel_get_help_uri (CcPanel *panel)
 {
-  if (!g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "Unity"))
-    return "help:ubuntu-help/prefs-display";
-  else
-    return "help:gnome-help/prefs-display";
+  return "help:gnome-help/prefs-display";
 }
 
 static void
--- a/panels/mouse/cc-mouse-panel.c
+++ b/panels/mouse/cc-mouse-panel.c
@@ -107,10 +107,7 @@
 static const char *
 cc_mouse_panel_get_help_uri (CcPanel *panel)
 {
-  if (!g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "Unity"))
-    return "help:ubuntu-help/mouse";
-  else
-    return "help:gnome-help/mouse";
+  return "help:gnome-help/mouse";
 }
 
 static void
--- a/panels/network/cc-network-panel.c
+++ b/panels/network/cc-network-panel.c
@@ -242,10 +242,7 @@
 static const char *
 cc_network_panel_get_help_uri (CcPanel *panel)
 {
-        if (!g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "Unity"))
-            return "help:ubuntu-help/net";
-        else
-            return "help:gnome-help/net";
+	return "help:gnome-help/net";
 }
 
 static void
--- a/panels/power/cc-power-panel.c
+++ b/panels/power/cc-power-panel.c
@@ -130,10 +130,7 @@
 static const char *
 cc_power_panel_get_help_uri (CcPanel *panel)
 {
-  if (!g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "Unity"))
-    return "help:ubuntu-help/power";
-  else
-    return "help:gnome-help/power";
+  return "help:gnome-help/power";
 }
 
 static void
--- a/panels/printers/cc-printers-panel.c
+++ b/panels/printers/cc-printers-panel.c
@@ -251,10 +251,7 @@
 static const char *
 cc_printers_panel_get_help_uri (CcPanel *panel)
 {
-  if (!g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "Unity"))
-    return "help:ubuntu-help/printing";
-  else
-    return "help:gnome-help/printing";
+  return "help:gnome-help/printing";
 }
 
 static void
--- a/panels/screen/cc-screen-panel.c
+++ b/panels/screen/cc-screen-panel.c
@@ -148,10 +148,7 @@
 static const char *
 cc_screen_panel_get_help_uri (CcPanel *panel)
 {
-  if (!g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "Unity"))
-    return "help:ubuntu-help/prefs-display";
-  else
-    return "help:gnome-help/prefs-display";
+  return "help:gnome-help/prefs-display";
 }
 
 static void
--- a/panels/universal-access/cc-ua-panel.c
+++ b/panels/universal-access/cc-ua-panel.c
@@ -159,10 +159,7 @@
 static const char *
 cc_ua_panel_get_help_uri (CcPanel *panel)
 {
-  if (!g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "Unity"))
-    return "help:ubuntu-help/a11y";
-  else
-    return "help:gnome-help/a11y";
+  return "help:gnome-help/a11y";
 }
 
 static void
--- a/panels/user-accounts/um-password-dialog.c
+++ b/panels/user-accounts/um-password-dialog.c
@@ -615,14 +615,9 @@
 
         widget = (GtkWidget *) gtk_builder_get_object (builder, "password-normal-strength-hints-label");
         old_label = gtk_label_get_label (GTK_LABEL (widget));
-        if (!g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "Unity"))
-            label = g_strdup_printf ("<a href=\"%s\">%s</a>",
-                                 "help:ubuntu-help/user-goodpassword",
-                                  old_label);
-        else
-            label = g_strdup_printf ("<a href=\"%s\">%s</a>",
-                                  "help:gnome-help/user-goodpassword",
-                                  old_label);
+        label = g_strdup_printf ("<a href=\"%s\">%s</a>",
+                                 "help:gnome-help/user-goodpassword",
+                                 old_label);
         gtk_label_set_markup (GTK_LABEL (widget), label);
         g_free (label);
 
--- a/panels/user-accounts/um-user-panel.c
+++ b/panels/user-accounts/um-user-panel.c
@@ -1536,10 +1536,7 @@
 static const char *
 cc_user_panel_get_help_uri (CcPanel *panel)
 {
- if (!g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "Unity"))
-   return "help:ubuntu-help/user-accounts";
- else
- return "help:gnome-help/user-accounts";
+	return "help:gnome-help/user-accounts";
 }
 
 static void
--- a/panels/wacom/cc-wacom-panel.c
+++ b/panels/wacom/cc-wacom-panel.c
@@ -124,10 +124,7 @@
 static const char *
 cc_wacom_panel_get_help_uri (CcPanel *panel)
 {
-  if (!g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "Unity"))
-    return "help:ubuntu-help/wacom";
-  else
-    return "help:gnome-help/wacom";
+  return "help:gnome-help/wacom";
 }
 
 static void
--- a/shell/control-center.c
+++ b/shell/control-center.c
@@ -181,12 +181,8 @@
 
   if (panel)
     uri = cc_panel_get_help_uri (panel);
-    if (!g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "Unity"))
-      gtk_show_uri (gtk_widget_get_screen (window),
-                uri ? uri : "help:ubuntu-help/prefs",
-                GDK_CURRENT_TIME, NULL);
-    else
-      gtk_show_uri (gtk_widget_get_screen (window),
+
+  gtk_show_uri (gtk_widget_get_screen (window),
                 uri ? uri : "help:gnome-help/prefs",
                 GDK_CURRENT_TIME, NULL);
 }
@@ -199,14 +195,9 @@
   GnomeControlCenter *shell = user_data;
   GtkWidget *window = cc_shell_get_toplevel (CC_SHELL (shell));
   
-  if (!g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "Unity"))
-    gtk_show_uri (gtk_widget_get_screen (window),
-                  "help:ubuntu-help/prefs",
-                  GDK_CURRENT_TIME, NULL);
-  else
-    gtk_show_uri (gtk_widget_get_screen (window),
-                  "help:gnome-help/prefs",
-                  GDK_CURRENT_TIME, NULL);
+  gtk_show_uri (gtk_widget_get_screen (window),
+                "help:gnome-help/prefs",
+                GDK_CURRENT_TIME, NULL);
 }
 
 static void
