diff -Nru gnome-disk-utility-3.0.2.orig/Makefile.am gnome-disk-utility-3.0.2/Makefile.am
--- gnome-disk-utility-3.0.2.orig/Makefile.am	2012-05-01 15:06:45.480319030 -0400
+++ gnome-disk-utility-3.0.2/Makefile.am	2012-05-01 15:07:03.533554561 -0400
@@ -1,6 +1,6 @@
 ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
 
-SUBDIRS = src data doc help po
+SUBDIRS = src
 
 EXTRA_DIST = \
 	autogen.sh		\
diff -Nru gnome-disk-utility-3.0.2.orig/src/Makefile.am gnome-disk-utility-3.0.2/src/Makefile.am
--- gnome-disk-utility-3.0.2.orig/src/Makefile.am	2012-05-01 15:06:45.446985878 -0400
+++ gnome-disk-utility-3.0.2/src/Makefile.am	2012-05-01 15:06:52.473614479 -0400
@@ -1,4 +1,4 @@
-SUBDIRS = gdu gdu-gtk palimpsest notification format-tool
+SUBDIRS = gdu
 
 if ENABLE_NAUTILUS
 SUBDIRS += nautilus-extension
