diff -Nru indicator-application-0.4.0.orig/src/indicator-application.c indicator-application-0.4.0/src/indicator-application.c
--- indicator-application-0.4.0.orig/src/indicator-application.c	2012-02-04 22:35:07.512976751 -0500
+++ indicator-application-0.4.0/src/indicator-application.c	2012-02-04 22:35:53.792978921 -0500
@@ -32,11 +32,7 @@
 #include <gtk/gtk.h>
 
 /* DBus Stuff */
-#ifdef HAVE_GTK3
-#include <libdbusmenu-gtk3/menu.h>
-#else
 #include <libdbusmenu-gtk/menu.h>
-#endif
 
 /* Indicator Stuff */
 #include <libindicator/indicator.h>
