From 3181e37bd0463262a66b4c0274bc629c9a524554 Mon Sep 17 00:00:00 2001
From: Xiao-Long Chen <chenxiaolong@cxl.epac.to>
Date: Thu, 10 Apr 2014 20:06:38 -0400
Subject: [PATCH 1/6] Disable tests

---
 CMakeLists.txt | 49 -------------------------------------------------
 1 file changed, 49 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c60ebdc..1477461 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -276,56 +276,7 @@ endif ()
 # subdirs
 add_subdirectory(doc)
 
-# Check for google test and build it locally
-set (
-  GMOCK_ROOT_DIR
-  "/usr/src/gmock"
-  CACHE
-  PATH
-  "Path to Google mock srcs"
-)
-
-set (
-  GMOCK_INCLUDE_DIR
-  "/usr/include/gmock/include"
-  CACHE
-  PATH
-  "Path to Google mock include"
-)
-
-set(
-  GTEST_ROOT_DIR
-  "${GMOCK_ROOT_DIR}/gtest" # Default value, adjustable by user with e.g., ccmake
-  CACHE
-  PATH
-  "Path to Google test srcs"
-)
-
-set(
-  GTEST_INCLUDE_DIR
-  "${GMOCK_ROOT_DIR}/gtest/include"
-  CACHE
-  PATH
-  "Path to Google tests include"
-)
-
-if (GTEST_INCLUDE_DIR AND GMOCK_INCLUDE_DIR)
-  include_directories (
-    ${GMOCK_INCLUDE_DIR}
-    ${GTEST_INCLUDE_DIR}
-  )
-  add_subdirectory(
-    ${GMOCK_ROOT_DIR}
-    "${CMAKE_CURRENT_BINARY_DIR}/gmock"
-  )
-elseif (GTEST_INCLUDE_DIR AND NOT GMOCK_INCLUDE_DIR)
-  include_directories (
-    ${GTEST_INCLUDE_DIR}
-  )
-endif(GTEST_INCLUDE_DIR AND GMOCK_INCLUDE_DIR)
-
 add_subdirectory(services)
-add_subdirectory(tests)
 add_subdirectory(tools)
 add_subdirectory(UnityCore)
 add_subdirectory(guides)
-- 
1.9.2

