From 3364fa4cf7581f7ae8ea8006e68f7a3440b2a98d Mon Sep 17 00:00:00 2001
From: Xiao-Long Chen <chenxiaolong@cxl.epac.to>
Date: Thu, 10 Apr 2014 20:10:51 -0400
Subject: [PATCH 3/8] Disable -Werror

---
 CMakeLists.txt          | 2 +-
 services/CMakeLists.txt | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0a099ed..94fde93 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -193,7 +193,7 @@ if (BOOT_LOGGER)
     SET (BOOT_LOGGER_FLAG "-DENABLE_LOGGER")
 endif (BOOT_LOGGER)
 
-SET (MAINTAINER_CXXFLAGS "-Werror -Wall -Wcast-align -Wempty-body -Wformat-security -Winit-self -Warray-bounds")
+SET (MAINTAINER_CXXFLAGS "-Wall -Wcast-align -Wempty-body -Wformat-security -Winit-self -Warray-bounds")
 
 option (DISABLE_ERROR_ON_LOCAL_TYPEDEFS_WARNINGS "Disable errors when local typedefs are unused" ON)
 if (DISABLE_ERROR_ON_LOCAL_TYPEDEFS_WARNINGS)
diff --git a/services/CMakeLists.txt b/services/CMakeLists.txt
index b80ec71..df0f767 100644
--- a/services/CMakeLists.txt
+++ b/services/CMakeLists.txt
@@ -35,7 +35,7 @@ set(PANEL_SOURCES
 set(CFLAGS
     ${SERVICE_DEPS_CFLAGS}
     ${SERVICE_DEPS_CFLAGS_OTHER}
-    "-Werror -Wall"
+    "-Wall"
     )
 
 string (REPLACE ";" " " CFLAGS "${CFLAGS}")
-- 
2.3.7

