diff -Nru unity-6.4.0.orig/plugins/unityshell/CMakeLists.txt unity-6.4.0/plugins/unityshell/CMakeLists.txt
--- unity-6.4.0.orig/plugins/unityshell/CMakeLists.txt	2012-09-14 20:05:32.454489830 -0400
+++ unity-6.4.0/plugins/unityshell/CMakeLists.txt	2012-09-14 20:09:54.231550070 -0400
@@ -10,8 +10,9 @@
                CFLAGSADD "-DINSTALLPREFIX='\"${CMAKE_INSTALL_PREFIX}\"' -DPKGDATADIR='\"${PKGDATADIR}\"' -I${CMAKE_BINARY_DIR} -I${CMAKE_SOURCE_DIR} ${BOOT_LOGGER_FLAG} -DGETTEXT_PACKAGE='\"unity\"' ${MAINTAINER_CFLAGS} -I${CMAKE_SOURCE_DIR}/dash/ -I${CMAKE_SOURCE_DIR}/launcher/ -I${CMAKE_SOURCE_DIR}/hud/ -I${CMAKE_SOURCE_DIR}/panel/ -I${CMAKE_SOURCE_DIR}/shortcuts/ -I${CMAKE_SOURCE_DIR}/unity-shared/"
                LIBDIRS "${CMAKE_BINARY_DIR}/UnityCore"
                )
-add_dependencies(unityshell unity-core-${UNITY_API_VERSION} dash-lib launcher-lib switcher-lib hud-lib panel-lib shortcuts-lib unity-shared unity-shared-compiz)
-target_link_libraries(unityshell unity-core-${UNITY_API_VERSION} launcher-lib dash-lib switcher-lib hud-lib panel-lib shortcuts-lib unity-shared unity-shared-compiz)
+pkg_check_modules(CACHED_GMODULE REQUIRED gmodule-2.0)
+add_dependencies(unityshell unity-core-${UNITY_API_VERSION} dash-lib launcher-lib switcher-lib hud-lib panel-lib shortcuts-lib unity-shared unity-shared-compiz ${CACHED_GMODULE_LIBRARIES})
+target_link_libraries(unityshell unity-core-${UNITY_API_VERSION} launcher-lib dash-lib switcher-lib hud-lib panel-lib shortcuts-lib unity-shared unity-shared-compiz ${CACHED_GMODULE_LIBRARIES})
 set_target_properties(unityshell 
                       PROPERTIES INSTALL_RPATH "${CACHED_UNITY_PRIVATE_DEPS_LIBRARY_DIRS}"
                       INSTALL_RPATH_USE_LINK_PATH TRUE)
diff -Nru unity-6.4.0.orig/services/CMakeLists.txt unity-6.4.0/services/CMakeLists.txt
--- unity-6.4.0.orig/services/CMakeLists.txt	2012-09-14 20:05:32.483489836 -0400
+++ unity-6.4.0/services/CMakeLists.txt	2012-09-14 20:06:00.528496283 -0400
@@ -43,7 +43,9 @@
     )
 add_definitions(${CFLAGS})
 
-set(LIBS ${SERVICE_DEPS_LIBRARIES})
+pkg_check_modules(CACHED_GMODULE REQUIRED gmodule-2.0)
+
+set(LIBS ${SERVICE_DEPS_LIBRARIES} ${CACHED_GMODULE_LIBRARIES})
 link_libraries(${LIBS})
 
 set(LIB_PATHS ${SERVICE_DEPS_LIBRARY_DIRS})
