From 04349cdcf0577f576f9078e5a790f8e7d81d4bb2 Mon Sep 17 00:00:00 2001
From: Xiao-Long Chen <chenxiaolong@cxl.epac.to>
Date: Wed, 24 Dec 2014 17:44:55 -0500
Subject: [PATCH 8/8] Fix missing include for shared_ptr

---
 unity-shared/DecorationStyle.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/unity-shared/DecorationStyle.h b/unity-shared/DecorationStyle.h
index 47a6885..c331f03 100644
--- a/unity-shared/DecorationStyle.h
+++ b/unity-shared/DecorationStyle.h
@@ -23,6 +23,7 @@
 #include <NuxCore/Property.h>
 #include <NuxCore/Rect.h>
 #include <cairo/cairo.h>
+#include <memory>
 
 struct _GtkStyleContext;
 
-- 
2.3.7

