--- a/panel/PanelMenuView.cpp
+++ b/panel/PanelMenuView.cpp
@@ -73,7 +73,7 @@
   , new_app_menu_shown_(false)
   , monitor_(0)
   , active_xid_(0)
-  , desktop_name_(_("Ubuntu Desktop"))
+  , desktop_name_(_("Arch Linux Desktop"))
   , menus_fadein_(DEFAULT_MENUS_FADEIN)
   , menus_fadeout_(DEFAULT_MENUS_FADEOUT)
   , menus_discovery_(DEFAULT_MENUS_DISCOVERY)
--- a/tests/autopilot/unity/tests/launcher/test_keynav.py
+++ b/tests/autopilot/unity/tests/launcher/test_keynav.py
@@ -148,7 +148,7 @@
         self.launcher_instance.key_nav_activate()
         self.addCleanup(self.keybinding, "expo/cancel")
 
-        self.assertThat(self.panels.get_active_panel().title, Eventually(Equals("Ubuntu Desktop")))
+        self.assertThat(self.panels.get_active_panel().title, Eventually(Equals("Arch Linux Desktop")))
 
     def test_launcher_keynav_expo_exit_on_esc(self):
         """Esc should quit expo when entering it from KeyNav."""
