diff -Nru unity-5.6.0.orig/CMakeLists.txt unity-5.6.0/CMakeLists.txt
--- unity-5.6.0.orig/CMakeLists.txt	2012-03-13 15:33:47.720430808 -0400
+++ unity-5.6.0/CMakeLists.txt	2012-03-13 15:34:09.557100015 -0400
@@ -134,10 +134,6 @@
 add_subdirectory(doc)
 
 find_path(GTEST_INCLUDE_DIR gtest/gtest.h)
-if (GTEST_INCLUDE_DIR)
-  #FIXME - hardcoded is bad!
-  add_subdirectory(/usr/src/gtest gtest)
-endif(GTEST_INCLUDE_DIR)
 
 add_subdirectory(services)
 add_subdirectory(tests)
