diff -Nru unity-5.8.0.orig/CMakeLists.txt unity-5.8.0/CMakeLists.txt
--- unity-5.8.0.orig/CMakeLists.txt	2012-03-25 02:23:19.804589414 -0400
+++ unity-5.8.0/CMakeLists.txt	2012-03-25 02:24:08.087923835 -0400
@@ -133,23 +133,7 @@
 # subdirs
 add_subdirectory(doc)
 
-# Check for google test and build it locally
-set(
-  GTEST_ROOT_DIR
-  "/usr/src/gtest" # Default value, adjustable by user with e.g., ccmake
-  CACHE
-  PATH
-  "Path to Google test srcs"
-)
-
 find_path(GTEST_INCLUDE_DIR gtest/gtest.h)
-if (GTEST_INCLUDE_DIR)
-  #FIXME - hardcoded is bad!
-  add_subdirectory(
-    ${GTEST_ROOT_DIR}
-    gtest
-  )
-endif(GTEST_INCLUDE_DIR)
 
 add_subdirectory(services)
 add_subdirectory(tests)
