From 978b54a62fff02c7adfc8794db75969869fb291b Mon Sep 17 00:00:00 2001
From: Xiao-Long Chen <chenxiaolong@cxl.epac.to>
Date: Sun, 13 Apr 2014 01:01:44 -0400
Subject: [PATCH 1/2] guest-account: Use cross-distro commands

---
 debian/guest-account | 45 +++++++++++++++++++++++++++++++++++----------
 1 file changed, 35 insertions(+), 10 deletions(-)

diff --git a/debian/guest-account b/debian/guest-account
index 47cfe70..77312ba 100644
--- a/debian/guest-account
+++ b/debian/guest-account
@@ -10,7 +10,12 @@
 
 add_account ()
 {
-  HOME=`mktemp -td guest-XXXXXX`
+  NEWHOME=`mktemp -td guest-XXXXXX`
+  # useradd does not accept upper case characters in the username
+  HOME=`echo $NEWHOME | tr '[A-Z]' '[a-z]'`
+  if [ "x$NEWHOME" != "x$HOME" ]; then
+    mv $NEWHOME $HOME
+  fi
   USER=`echo $HOME | sed 's/\(.*\)guest/guest/'`
 
   # if $USER already exists, it must be a locked system account with no existing
@@ -36,7 +41,21 @@ add_account ()
     fi
   else
     # does not exist, so create it
-    adduser --system --no-create-home --home / --gecos "Guest" --group --shell /bin/bash $USER || {
+    # Arch Linux modification: Ubuntu/Debian uses their own adduser package,
+    # which works differently from the own provided by the shadow package.
+    # Instead, use useradd, which works in any distro.
+    #
+    # Only the syntax is changed
+    # adduser          -> useradd
+    # --system         -> --system
+    # --no-create-home -> --no-create-home
+    # --home           -> --home-dir
+    # --gecos          -> --comment
+    # --group          -> --user-group
+    # --shell          -> --shell
+    #
+    #adduser --system --no-create-home --home / --gecos "Guest" --group --shell /bin/bash $USER || {
+    useradd --system --no-create-home --home-dir / --comment "Guest" --user-group --shell /bin/bash $USER || {
         umount "$HOME"
         rm -rf "$HOME"
         exit 1
@@ -115,10 +134,15 @@ remove_account ()
     echo "Error: invalid user $USER"
     exit 1
   }
-  UID=`echo "$PWENT" | cut -f3 -d:`
+  USERUID=`echo "$PWENT" | cut -f3 -d:`
   HOME=`echo "$PWENT" | cut -f6 -d:`
 
-  if [ "$UID" -ge 500 ]; then
+  # deluser is provided by the adduser package on Debian/Ubuntu. useradd
+  # doesn't have a '--system' parameter, which causes deluser to only delete
+  # system users, so this will be handled using this script.
+  SYS_UID_MIN="$(cat /etc/login.defs | grep SYS_UID_MIN | awk '{print $2}')"
+  SYS_UID_MAX="$(cat /etc/login.defs | grep SYS_UID_MAX | awk '{print $2}')"
+  if [ "$USERUID" -lt "$SYS_UID_MIN" ] || [ "$USERUID" -gt "$SYS_UID_MAX" ]; then
     echo "Error: user $USER is not a system user."
     exit 1
   fi
@@ -138,17 +162,18 @@ remove_account ()
   rm -rf "$HOME"
 
   # remove leftovers in /tmp
-  find /tmp -mindepth 1 -maxdepth 1 -uid "$UID" -print0 | xargs -0 rm -rf || true
+  find /tmp -mindepth 1 -maxdepth 1 -uid "$USERUID" -print0 | xargs -0 rm -rf || true
 
-  # remove possible /media/guest-XXXXXX folder
-  if [ -d /media/"$USER" ]; then
-    for dir in $( find /media/"$USER" -mindepth 1 -maxdepth 1 ); do
+  # remove possible /run/media/guest-XXXXXX folder
+  if [ -d /run/media/"$USER" ]; then
+    for dir in $( find /run/media/"$USER" -mindepth 1 -maxdepth 1 ); do
       umount "$dir" || true
     done
-    rmdir /media/"$USER" || true
+    rmdir /run/media/"$USER" || true
   fi
 
-  deluser --system "$USER"
+  #deluser --system "$USER"
+  userdel "$USER"
 }
 
 case "$1" in
-- 
1.9.2

