From 3de0924902388870f4a9ac1e3bda625e23a78a83 Mon Sep 17 00:00:00 2001
From: Xiao-Long Chen <chenxiaolong@cxl.epac.to>
Date: Sun, 13 Apr 2014 01:02:42 -0400
Subject: [PATCH 2/2] guest-account: Add default GSettings support

---
 debian/guest-account | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/debian/guest-account b/debian/guest-account
index 77312ba..fadec17 100644
--- a/debian/guest-account
+++ b/debian/guest-account
@@ -121,6 +121,16 @@ add_account ()
       . /etc/guest-session/prefs.sh
   fi
 
+  # Load default dconf settings
+  mkdir -p "$HOME/.config/" "$HOME/.cache/"
+  chown -R $USER:$USER "$HOME/.config/" "$HOME/.cache/"
+  if [ -d /etc/guest-session/gsettings/ ]; then
+    for i in /etc/guest-session/gsettings/*; do
+      cat "${i}" | sed '/^\[.*\]$/ s/\./\//g' | \
+        su - $USER -c 'dbus-launch dconf load /'
+    done
+  fi
+
   chown -R $USER:$USER "$HOME"
 
   echo $USER  
-- 
1.9.2

