--- a/configure.ac
+++ b/configure.ac
@@ -19,8 +19,7 @@
 URL_DISPATCHER_1_REQUIRED_VERSION=1
 
 PKG_CHECK_MODULES(SERVICE_DEPS, [glib-2.0 >= $GLIB_REQUIRED_VERSION
-                                 gio-unix-2.0 >= $GIO_REQUIRED_VERSION
-                                 url-dispatcher-1 >= $URL_DISPATCHER_1_REQUIRED_VERSION])
+                                 gio-unix-2.0 >= $GIO_REQUIRED_VERSION])
 
 
 dnl ##############################
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -18,7 +18,6 @@
 	--vapidir=./ \
 	--pkg config \
 	--pkg rfkill \
-	--pkg url-dispatcher \
 	--pkg posix \
 	--pkg glib-2.0 \
 	--pkg gio-2.0
--- a/src/phone.vala
+++ b/src/phone.vala
@@ -61,7 +61,7 @@
 
   void show_settings (string panel)
   {
-    UrlDispatch.send ("settings:///system/bluetooth");
+    //UrlDispatch.send ("settings:///system/bluetooth");
   }
 
   Action create_settings_action ()
--- a/vapi/url-dispatcher.vapi
+++ b/vapi/url-dispatcher.vapi
@@ -1,9 +0,0 @@
-[CCode (cprefix = "", lower_case_cprefix = "", cheader_filename = "liburl-dispatcher-1/url-dispatcher.h")]
-
-namespace UrlDispatch
-{
-  public delegate void DispatchCallback ();
-
-  [CCode (cname = "url_dispatch_send")]
-  public static void send (string url, DispatchCallback? func = null);
-}
