From a6ec4943a120c69f5f01bf1c8c15629dfd12f850 Mon Sep 17 00:00:00 2001
From: Xiao-Long Chen <chenxiaolong@cxl.epac.to>
Date: Mon, 7 Apr 2014 20:22:05 -0400
Subject: [PATCH 1/4] Fix connection issue due to untrusted SSL CA certificate

---
 src/smart-scopes-search.vala | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/smart-scopes-search.vala b/src/smart-scopes-search.vala
index 9e4fb08..eabe963 100644
--- a/src/smart-scopes-search.vala
+++ b/src/smart-scopes-search.vala
@@ -209,7 +209,8 @@ namespace Unity.HomeScope.SmartScopes {
         geo_store = Environment.get_variable ("SMART_SCOPES_GEO_STORE");
 
         session = new Soup.SessionAsync ();
-        session.ssl_use_system_ca_file = true;
+        session.ssl_use_system_ca_file = false;
+        session.ssl_ca_file = "/etc/ssl/certs/UbuntuOne-Go_Daddy_CA.pem";
         session.ssl_strict = true;
         session.user_agent = "Unity Home Scope v" + Config.VERSION;
         session.add_feature_by_type (typeof (SoupGNOME.ProxyResolverGNOME));
-- 
2.1.1

