--- a/tests/test-libappindicator-dbus-client.c
+++ b/tests/test-libappindicator-dbus-client.c
@@ -204,6 +204,7 @@
 gint
 main (gint argc, gchar * argv[])
 {
+	g_type_init();
 	GError * error = NULL;
 	DBusGConnection * session_bus = dbus_g_bus_get(DBUS_BUS_SESSION, &error);
 	if (error != NULL) {
--- a/tests/test-libappindicator-dbus-server.c
+++ b/tests/test-libappindicator-dbus-server.c
@@ -37,6 +37,7 @@
 gint
 main (gint argc, gchar * argv[])
 {
+	g_type_init();
 	gtk_init(&argc, &argv);
 
 	AppIndicator * ci = app_indicator_new (TEST_ID, TEST_ICON_NAME, TEST_CATEGORY);
--- a/tests/test-libappindicator-fallback-item.c
+++ b/tests/test-libappindicator-fallback-item.c
@@ -106,6 +106,7 @@
 int
 main (int argc, char ** argv)
 {
+	g_type_init();
 	gtk_init(&argc, &argv);
 
 	GError * error = NULL;
--- a/tests/test-libappindicator-fallback-watcher.c
+++ b/tests/test-libappindicator-fallback-watcher.c
@@ -58,6 +58,8 @@
 int
 main (int argv, char ** argc)
 {
+	g_type_init();
+
 	g_debug("Waiting to init.");
 
 
--- a/tests/test-libappindicator-status-client.c
+++ b/tests/test-libappindicator-status-client.c
@@ -109,6 +109,7 @@
 gint
 main (gint argc, gchar * argv[])
 {
+	g_type_init();
 	GError * error = NULL;
 	DBusGConnection * session_bus = dbus_g_bus_get(DBUS_BUS_SESSION, &error);
 	if (error != NULL) {
--- a/tests/test-libappindicator-status-server.c
+++ b/tests/test-libappindicator-status-server.c
@@ -60,6 +60,7 @@
 gint
 main (gint argc, gchar * argv[])
 {
+	g_type_init();
 	gtk_init(&argc, &argv);
 
 	g_usleep(100000);
