diff -Nru appmenu-gtk-12.10.3daily12.11.28.orig/configure.ac appmenu-gtk-12.10.3daily12.11.28/configure.ac
--- appmenu-gtk-12.10.3daily12.11.28.orig/configure.ac	2013-01-07 15:04:14.064030072 -0500
+++ appmenu-gtk-12.10.3daily12.11.28/configure.ac	2013-01-07 15:04:24.140643763 -0500
@@ -5,7 +5,7 @@
 AC_CONFIG_SRCDIR([src/bridge.c])
 AC_CONFIG_MACRO_DIR([m4])
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADER(config.h)
 AM_INIT_AUTOMAKE([1.9])
 
 AM_MAINTAINER_MODE
