From aeda0048c1efca132b7ed9f75f5f0e4625819378 Mon Sep 17 00:00:00 2001
From: Xiao-Long Chen <chenxiaolong@cxl.epac.to>
Date: Thu, 30 Oct 2014 13:27:22 -0400
Subject: [PATCH] Fix the autostart command

---
 data/nautilus-autostart.desktop.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/nautilus-autostart.desktop.in b/data/nautilus-autostart.desktop.in
index 7252f87..5dabcaf 100644
--- a/data/nautilus-autostart.desktop.in
+++ b/data/nautilus-autostart.desktop.in
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Type=Application
 Name=Files
-Exec=nautilus -n
+Exec=nautilus --gapplication-service
 OnlyShowIn=GNOME;Unity;
 AutostartCondition=GSettings org.gnome.desktop.background show-desktop-icons
-- 
2.1.3

