diff -Nru unity-lens-files-5.10.0.orig/src/folder.vala unity-lens-files-5.10.0/src/folder.vala
--- unity-lens-files-5.10.0.orig/src/folder.vala	2012-04-26 12:06:39.598833926 -0400
+++ unity-lens-files-5.10.0/src/folder.vala	2012-04-26 12:43:44.623446599 -0400
@@ -208,7 +208,7 @@
   
   }
   
-  private class Bookmark : Object
+  public class Bookmark : Object
   {
     public string uri { get; set construct; }
     public string icon { get; set construct; }
diff -Nru unity-lens-files-5.10.0.orig/src/utils.vala unity-lens-files-5.10.0/src/utils.vala
--- unity-lens-files-5.10.0.orig/src/utils.vala	2012-04-26 12:06:39.598833926 -0400
+++ unity-lens-files-5.10.0/src/utils.vala	2012-04-26 12:24:33.113018201 -0400
@@ -98,9 +98,8 @@
     
     public int cmp_file_info_by_mtime (FileInfo info1, FileInfo info2)
     {
-      TimeVal tv1, tv2;
-      info1.get_modification_time (out tv1);
-      info2.get_modification_time (out tv2);
+      TimeVal tv1 = info1.get_modification_time ();
+      TimeVal tv2 = info2.get_modification_time ();
       long cmp = tv1.tv_sec - tv2.tv_sec;
       return cmp < 0 ? 1 : ( cmp > 0 ? -1 : 0);
     }
