--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -105,12 +105,6 @@
 find_package(Qt5Test REQUIRED)
 include_directories(${Qt5Test_INCLUDE_DIRS})
 
-pkg_check_modules(QTDBUSTEST REQUIRED libqtdbustest-1 REQUIRED)
-include_directories(${QTDBUSTEST_INCLUDE_DIRS})
-
-pkg_check_modules(QTDBUSMOCK REQUIRED libqtdbusmock-1 REQUIRED)
-include_directories(${QTDBUSMOCK_INCLUDE_DIRS})
-
 pkg_check_modules(UNITYVOICE REQUIRED libunityvoice-1 REQUIRED)
 include_directories(${UNITYVOICE_INCLUDE_DIRS})
 
@@ -171,8 +165,6 @@
 	)
 endif()
 
-add_subdirectory(tests)
-
 add_subdirectory(tools-vala)
 if(${ENABLE_DOCUMENTATION})
   add_subdirectory(docs)
