diff -Nru indicator-session-12.10.2.orig/src/indicator-session.c indicator-session-12.10.2/src/indicator-session.c
--- indicator-session-12.10.2.orig/src/indicator-session.c	2012-09-09 13:35:42.659540407 -0400
+++ indicator-session-12.10.2/src/indicator-session.c	2012-09-09 13:44:10.077463822 -0400
@@ -447,7 +447,10 @@
     a11y = g_strdup (_("System"));
 
   g_debug (G_STRLOC" setting a11y to \"%s\"", a11y);
-  g_clear_pointer (&indicator->entry.accessible_desc, g_free);
+  if (indicator->entry.accessible_desc != NULL) {
+    g_free (indicator->entry.accessible_desc);
+    indicator->entry.accessible_desc = NULL;
+  }
   indicator->entry.accessible_desc = a11y;
   g_signal_emit (indicator,
                  INDICATOR_OBJECT_SIGNAL_ACCESSIBLE_DESC_UPDATE_ID,
diff -Nru indicator-session-12.10.2.orig/src/session-dbus.c indicator-session-12.10.2/src/session-dbus.c
--- indicator-session-12.10.2.orig/src/session-dbus.c	2012-09-09 13:35:42.654540419 -0400
+++ indicator-session-12.10.2/src/session-dbus.c	2012-09-09 13:43:09.356593727 -0400
@@ -206,7 +206,10 @@
 {
 	SessionDbusPrivate * priv = SESSION_DBUS_GET_PRIVATE(object);
 
-	g_clear_pointer (&priv->name, g_free);
+	if (priv->name != NULL) {
+		g_free(priv->name);
+		priv->name = NULL;
+	}
 
 	G_OBJECT_CLASS (session_dbus_parent_class)->finalize (object);
 	return;
diff -Nru indicator-session-12.10.2.orig/src/users-service-dbus.c indicator-session-12.10.2/src/users-service-dbus.c
--- indicator-session-12.10.2.orig/src/users-service-dbus.c	2012-09-09 13:35:42.656540415 -0400
+++ indicator-session-12.10.2/src/users-service-dbus.c	2012-09-09 13:43:09.356593727 -0400
@@ -735,7 +735,11 @@
   if (!g_strcmp0 (ssid, priv->guest_ssid))
     {
       g_debug ("%s removing guest session %s", G_STRLOC, ssid);
-      g_clear_pointer (&priv->guest_ssid, g_free);
+      if (priv->guest_ssid != NULL) {
+        g_free(priv->guest_ssid);
+        priv->guest_ssid = NULL;
+      }
+
       emit_guest_login_changed (service);
     }
   else
