diff -Nru indicator-session-12.10.4.orig/configure.ac indicator-session-12.10.4/configure.ac
--- indicator-session-12.10.4.orig/configure.ac	2013-01-10 05:20:14.001951378 -0500
+++ indicator-session-12.10.4/configure.ac	2013-01-10 05:20:22.595238808 -0500
@@ -3,7 +3,7 @@
 
 AC_PREREQ(2.53)
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AM_INIT_AUTOMAKE(indicator-session, 12.10.4)
 
 AM_MAINTAINER_MODE
