=== modified file 'src/session-menu-mgr.c'
--- src/session-menu-mgr.c	2013-03-06 15:29:26 +0000
+++ src/session-menu-mgr.c	2013-03-23 18:19:30 +0000
@@ -1280,9 +1280,9 @@
 static gboolean
 is_this_guest_session (void)
 {
-  /* FIXME: this test has been here awhile and seems to work,
-     but seems brittle to me */
-  return geteuid() < 500;
+  /* Guest accounts are created as system accounts, which have a UID less than
+     1000 in Arch Linux */
+  return geteuid() < 1000;
 }
 
 static gboolean
