diff -Nru indicator-session-0.3.7.1.orig/src/apt-watcher.h indicator-session-0.3.7.1/src/apt-watcher.h
--- indicator-session-0.3.7.1.orig/src/apt-watcher.h	2012-02-03 20:02:09.761353647 -0500
+++ indicator-session-0.3.7.1/src/apt-watcher.h	2012-02-03 20:02:45.044688635 -0500
@@ -25,11 +25,7 @@
 #include <libdbusmenu-glib/client.h>
 
 #include <gtk/gtk.h>
-#if GTK_CHECK_VERSION(3, 0, 0)
-#include <libdbusmenu-gtk3/menuitem.h>
-#else
 #include <libdbusmenu-gtk/menuitem.h>
-#endif
 
 #include "session-dbus.h"
 
diff -Nru indicator-session-0.3.7.1.orig/src/device-menu-mgr.c indicator-session-0.3.7.1/src/device-menu-mgr.c
--- indicator-session-0.3.7.1.orig/src/device-menu-mgr.c	2012-02-03 20:02:09.764686981 -0500
+++ indicator-session-0.3.7.1/src/device-menu-mgr.c	2012-02-03 20:03:52.304691786 -0500
@@ -18,7 +18,7 @@
 */
 
 #include <libdbusmenu-glib/client.h>
-#include <libdbusmenu-gtk3/menuitem.h>
+#include <libdbusmenu-gtk/menuitem.h>
 
 #include "device-menu-mgr.h"
 #include "settings-helper.h"
diff -Nru indicator-session-0.3.7.1.orig/src/indicator-session.c indicator-session-0.3.7.1/src/indicator-session.c
--- indicator-session-0.3.7.1.orig/src/indicator-session.c	2012-02-03 20:02:09.764686981 -0500
+++ indicator-session-0.3.7.1/src/indicator-session.c	2012-02-03 20:03:17.944690175 -0500
@@ -31,11 +31,7 @@
 #include <gtk/gtk.h>
 #include <gio/gio.h>
 
-#if GTK_CHECK_VERSION(3, 0, 0)
-#include <libdbusmenu-gtk3/menu.h>
-#else
 #include <libdbusmenu-gtk/menu.h>
-#endif
 
 #include <libindicator/indicator.h>
 #include <libindicator/indicator-object.h>
diff -Nru indicator-session-0.3.7.1.orig/src/session-service.c indicator-session-0.3.7.1/src/session-service.c
--- indicator-session-0.3.7.1.orig/src/session-service.c	2012-02-03 20:02:09.761353647 -0500
+++ indicator-session-0.3.7.1/src/session-service.c	2012-02-03 20:03:07.171356336 -0500
@@ -39,11 +39,7 @@
 #include <libdbusmenu-glib/client.h>
 
 #include <gtk/gtk.h>
-#if GTK_CHECK_VERSION(3, 0, 0)
-#include <libdbusmenu-gtk3/menuitem.h>
-#else
 #include <libdbusmenu-gtk/menuitem.h>
-#endif
 
 #include <libindicator/indicator-service.h>
 
diff -Nru indicator-session-0.3.7.1.orig/src/udev-mgr.h indicator-session-0.3.7.1/src/udev-mgr.h
--- indicator-session-0.3.7.1.orig/src/udev-mgr.h	2012-02-03 20:02:09.761353647 -0500
+++ indicator-session-0.3.7.1/src/udev-mgr.h	2012-02-03 20:02:54.954689099 -0500
@@ -24,11 +24,7 @@
 #include <libdbusmenu-glib/client.h>
 
 #include <gtk/gtk.h>
-#if GTK_CHECK_VERSION(3, 0, 0)
-#include <libdbusmenu-gtk3/menuitem.h>
-#else
 #include <libdbusmenu-gtk/menuitem.h>
-#endif
 
 G_BEGIN_DECLS
 
diff -Nru indicator-session-0.3.7.1.orig/src/user-menu-mgr.h indicator-session-0.3.7.1/src/user-menu-mgr.h
--- indicator-session-0.3.7.1.orig/src/user-menu-mgr.h	2012-02-03 20:02:09.764686981 -0500
+++ indicator-session-0.3.7.1/src/user-menu-mgr.h	2012-02-03 20:03:38.414691136 -0500
@@ -23,7 +23,7 @@
 
 
 #include <glib-object.h>
-#include <libdbusmenu-gtk3/menuitem.h>
+#include <libdbusmenu-gtk/menuitem.h>
 
 #include "session-dbus.h"
 
diff -Nru indicator-session-0.3.7.1.orig/src/user-widget.h indicator-session-0.3.7.1/src/user-widget.h
--- indicator-session-0.3.7.1.orig/src/user-widget.h	2012-02-03 20:02:09.764686981 -0500
+++ indicator-session-0.3.7.1/src/user-widget.h	2012-02-03 20:04:02.184692249 -0500
@@ -20,11 +20,7 @@
 #define __USER_WIDGET_H__
 
 #include <gtk/gtk.h>
-#if GTK_CHECK_VERSION(3, 0, 0)
-#include <libdbusmenu-gtk3/menuitem.h>
-#else
 #include <libdbusmenu-gtk/menuitem.h>
-#endif
 
 G_BEGIN_DECLS
 
