diff -Nru activity-log-manager-0.9.4.orig/Makefile.am activity-log-manager-0.9.4/Makefile.am
--- activity-log-manager-0.9.4.orig/Makefile.am	2012-05-03 14:59:41.729188379 -0400
+++ activity-log-manager-0.9.4/Makefile.am	2012-05-03 15:00:23.865626774 -0400
@@ -7,7 +7,7 @@
                  intltool-merge.in \
                  intltool-update.in
 
-almdocdir = ${prefix}/doc/alm
+almdocdir = ${prefix}/share/doc/alm
 almdoc_DATA = \
 	README\
 	COPYING\
