--- a/data/com.canonical.Unity.Lenses.gschema.xml.in.in
+++ b/data/com.canonical.Unity.Lenses.gschema.xml.in.in
@@ -2,10 +2,10 @@
 <schemalist>
   <schema path="/com/canonical/unity/lenses/" id="com.canonical.Unity.Lenses" gettext-domain="libunity">
     <key type="s" name="remote-content-search">
-      <default>'all'</default>
+      <default>'none'</default>
       <choices><choice value='all'/><choice value='none'/></choices>
       <summary>Content fetching from remote source preference.</summary>
-      <description>"all" is to enable the supported default lens to search from remote and commercial sources. "none" will indicate the lenses to not perform that remote search at all.</description>
+      <description>"all" is to enable the supported connonical default lens to search from remote and commercial sources. "none" will indicate the lenses to not perform that remote search at all.</description>
     </key>
     <key type="as" name="always-search">
       <default>['applications.scope','music.scope','videos.scope','files.scope']</default>
