diff -Nru libunity-7.1.4+14.10.20140808.orig/tools/music-track-model-renderer.vala libunity-7.1.4+14.10.20140808/tools/music-track-model-renderer.vala
--- libunity-7.1.4+14.10.20140808.orig/tools/music-track-model-renderer.vala	2015-04-26 20:37:40.947896956 +0200
+++ libunity-7.1.4+14.10.20140808/tools/music-track-model-renderer.vala	2015-04-26 20:49:56.206756603 +0200
@@ -27,13 +27,13 @@
     private Dee.ModelTag<int> track_model_tag;
     private int row_counter = 0;
     private ulong model_sync_sig_id = 0;
-    public ListStore track_view_model { get; construct; }
+    public Gtk.ListStore track_view_model { get; construct; }
 
     public signal void track_list_synchronized();
 
     construct
     {
-        track_view_model = new ListStore(6, typeof(string), typeof(int), typeof(string), typeof(uint), typeof(uint), typeof(double));
+        track_view_model = new Gtk.ListStore(6, typeof(string), typeof(int), typeof(string), typeof(uint), typeof(uint), typeof(double));
     }
 
     public MusicTrackModelRenderer(Dee.Model track_model)
diff -Nru libunity-7.1.4+14.10.20140808.orig/tools/unity-tool-ui.vala libunity-7.1.4+14.10.20140808/tools/unity-tool-ui.vala
--- libunity-7.1.4+14.10.20140808.orig/tools/unity-tool-ui.vala	2015-04-26 20:37:40.947896956 +0200
+++ libunity-7.1.4+14.10.20140808/tools/unity-tool-ui.vala	2015-04-26 20:49:56.210090049 +0200
@@ -40,9 +40,9 @@
             window.destroy.connect(Gtk.main_quit);
             window.show_all();
 
-            uimodel = builder.get_object("results_model") as ListStore;
-            ui_filter_model = builder.get_object("filters_model") as ListStore;
-            ui_cat_model = builder.get_object("categories_model") as ListStore;
+            uimodel = builder.get_object("results_model") as Gtk.ListStore;
+            ui_filter_model = builder.get_object("filters_model") as Gtk.ListStore;
+            ui_cat_model = builder.get_object("categories_model") as Gtk.ListStore;
 
             notebook = builder.get_object("notebook") as Notebook;
             search_entry = builder.get_object("search_entry") as Entry;
@@ -155,7 +155,7 @@
             builder.add_from_resource("/com/canonical/Unity/unity-tool/dbus-scope-connect.ui");
             builder.connect_signals(this);
             scope_discovery_spinner = builder.get_object("scope_discovery_spinner") as Spinner;
-            scope_list_model = builder.get_object("scope_list_model") as ListStore;
+            scope_list_model = builder.get_object("scope_list_model") as Gtk.ListStore;
             scope_list_combobox = builder.get_object("scope_list_combobox") as ComboBox;
             dbus_name_entry = builder.get_object("dbus_name_entry") as Entry;
             dbus_path_entry = builder.get_object("dbus_path_entry") as Entry;
@@ -843,10 +843,10 @@
     private Gtk.Entry dbus_name_entry = null;
     private Gtk.Entry dbus_path_entry = null;
     private Gtk.ComboBox scope_list_combobox = null;
-    private ListStore uimodel = null;
-    private ListStore ui_filter_model = null;
-    private ListStore scope_list_model = null;
-    private ListStore ui_cat_model = null;
+    private Gtk.ListStore uimodel = null;
+    private Gtk.ListStore ui_filter_model = null;
+    private Gtk.ListStore scope_list_model = null;
+    private Gtk.ListStore ui_cat_model = null;
     private ulong model_sync_sig_id;
     private Dee.SharedModel? dee_results_model = null;
     private string current_channel_id = "";
