diff -Nru indicator-messages-0.5.0.orig/src/indicator-messages.c indicator-messages-0.5.0/src/indicator-messages.c
--- indicator-messages-0.5.0.orig/src/indicator-messages.c	2012-02-03 20:29:59.521431877 -0500
+++ indicator-messages-0.5.0/src/indicator-messages.c	2012-02-03 20:30:33.918100155 -0500
@@ -26,13 +26,8 @@
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
 
-#if GTK_CHECK_VERSION(3, 0, 0)
-#include <libdbusmenu-gtk3/menu.h>
-#include <libdbusmenu-gtk3/menuitem.h>
-#else
 #include <libdbusmenu-gtk/menu.h>
 #include <libdbusmenu-gtk/menuitem.h>
-#endif
 
 #include <libindicator/indicator.h>
 #include <libindicator/indicator-object.h>
